<!DOCTYPE html>



  


<html class="theme-next pisces use-motion" lang="zh-CN">
<head><meta name="generator" content="Hexo 3.9.0">
  <meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
<meta name="theme-color" content="#222">












<meta http-equiv="Cache-Control" content="no-transform">
<meta http-equiv="Cache-Control" content="no-siteapp">






















<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css">

<link href="/css/main.css?v=6.0.3" rel="stylesheet" type="text/css">


  <link rel="apple-touch-icon" sizes="180x180" href="/images/bear.png?v=6.0.3">


  <link rel="icon" type="image/png" sizes="32x32" href="/images/bear.png?v=6.0.3">


  <link rel="icon" type="image/png" sizes="16x16" href="/images/bear.png?v=6.0.3">










<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Pisces',
    version: '6.0.3',
    sidebar: {"position":"left","display":"post","offset":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: false,
    fastclick: false,
    lazyload: false,
    tabs: true,
    motion: {"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>


  




  
  <meta name="keywords" content="RNA-seq,">


<meta name="description" content="自学习RNA-seq以来，不断遇到困难再克服苦难，但是回顾整个过程，发现自己照着流程跑并没有真正理解这个过程每一步的意义，希望通过不过更新此blog，加深理解。">
<meta name="keywords" content="RNA-seq">
<meta property="og:type" content="article">
<meta property="og:title" content="理解RNA-seq分析流程">
<meta property="og:url" content="https://william-chen777.github.io/2020/09/17/理解RNA-seq分析流程/index.html">
<meta property="og:site_name" content="William Chen">
<meta property="og:description" content="自学习RNA-seq以来，不断遇到困难再克服苦难，但是回顾整个过程，发现自己照着流程跑并没有真正理解这个过程每一步的意义，希望通过不过更新此blog，加深理解。">
<meta property="og:locale" content="zh-CN">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917202134998.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917211852781.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917212134860.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917212446915.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917212618909.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917212736360.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917212902342.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917213048760.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917213134145.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200917213246762.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200918095138875.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200918095240207.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200918100400878.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920155413361.jpg#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920155826152.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920161715935.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920162352791.jpg#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920162420521.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920171629992.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920172015367.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:image" content="https://img-blog.csdnimg.cn/20200920174006455.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">
<meta property="og:updated_time" content="2020-09-20T09:52:04.011Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="理解RNA-seq分析流程">
<meta name="twitter:description" content="自学习RNA-seq以来，不断遇到困难再克服苦难，但是回顾整个过程，发现自己照着流程跑并没有真正理解这个过程每一步的意义，希望通过不过更新此blog，加深理解。">
<meta name="twitter:image" content="https://img-blog.csdnimg.cn/20200917202134998.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center">






  <link rel="canonical" href="https://william-chen777.github.io/2020/09/17/理解RNA-seq分析流程/">



<script type="text/javascript" id="page.configurations">
  CONFIG.page = {
    sidebar: "",
  };
</script>
  <title>理解RNA-seq分析流程 | William Chen</title>
  









  <noscript>
  <style type="text/css">
    .use-motion .motion-element,
    .use-motion .brand,
    .use-motion .menu-item,
    .sidebar-inner,
    .use-motion .post-block,
    .use-motion .pagination,
    .use-motion .comments,
    .use-motion .post-header,
    .use-motion .post-body,
    .use-motion .collection-title { opacity: initial; }

    .use-motion .logo,
    .use-motion .site-title,
    .use-motion .site-subtitle {
      opacity: initial;
      top: initial;
    }

    .use-motion {
      .logo-line-before i { left: initial; }
      .logo-line-after i { right: initial; }
    }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-CN">

  
  
    
  

  <div class="container sidebar-position-left page-post-detail">
    <div class="headband"></div>

    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"> <div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/" class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">William Chen</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            <i class="menu-item-icon fa fa-fw fa-home"></i> <br>首页</a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            <i class="menu-item-icon fa fa-fw fa-tags"></i> <br>标签</a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            <i class="menu-item-icon fa fa-fw fa-th"></i> <br>分类</a>
        </li>
      
        
        <li class="menu-item menu-item-archives">
          <a href="/archives/" rel="section">
            <i class="menu-item-icon fa fa-fw fa-archive"></i> <br>归档</a>
        </li>
      

      
    </ul>
  

  
</nav>


  



 </div>
    </header>

    


    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            

  <div id="posts" class="posts-expand">
    

  

  
  
  

  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="https://william-chen777.github.io/2020/09/17/理解RNA-seq分析流程/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Chen kun">
      <meta itemprop="description" content>
      <meta itemprop="image" content="/images/avatar.png">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="William Chen">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">理解RNA-seq分析流程</h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2020-09-17T20:18:23+08:00">2020-09-17</time>
            

            
            

            
          </span>

          
            <span class="post-category">
            
              <span class="post-meta-divider">|</span>
            
              <span class="post-meta-item-icon">
                <i class="fa fa-folder-o"></i>
              </span>
              
                <span class="post-meta-item-text">分类于</span>
              
              
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing"><a href="/categories/生物信息学/" itemprop="url" rel="index"><span itemprop="name">生物信息学</span></a></span>

                
                
              
            </span>
          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        <p>自学习RNA-seq以来，不断遇到困难再克服苦难，但是回顾整个过程，发现自己照着流程跑并没有真正理解这个过程每一步的意义，希望通过不过更新此blog，加深理解。</p>
<a id="more"></a>

<hr>
<h1 id="什么是高通量转录组测序？"><a href="#什么是高通量转录组测序？" class="headerlink" title="什么是高通量转录组测序？"></a>什么是高通量转录组测序？</h1><p><strong>什么是高通量转录组测序</strong><br>顾名思义就是<strong>通量很高</strong>（相对于Sanger测序）的测序，一次可以获得海量的数据，所以叫高通量测序。<br><br><br><strong>转录组是什么？</strong><br>转录组，一般指的是，某一时空条件下细胞所产生的所有转录产物，说人话就是，<strong>你的样品经过了某种处理，然后拿去提了总RNA，这个总RNA就是转录组</strong>。<br><br><br><strong>理解高通量转录组测序的关键在哪？</strong><br>首先是<strong>建库</strong>，我们建库用的是cDNA，那么为什么要用DNA而不是RNA呢？<br>除了<strong>单链RNA不稳定</strong>外，还有一部分原因就是DNA的建库流程已经确定了，只要把RNA变成DNA后面流程完全一样。不过为了节约时间可以在一二链合成的时候直接加好接头，后面就连接头都不用加了，缩短建库时间，一天可以轻松完成建库。<br><br><br><strong>什么是桥式PCR？</strong><br><img src="https://img-blog.csdnimg.cn/20200917202134998.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"><br>大概过程就是 <strong>序列接头（adapter）</strong> 一端被固定，然后另一端跟反应槽里的互补序列配对，呈现桥状，然后进行PCR，故而称为桥式PCR。<br>经过n轮桥式PCR之后，一个序列可以扩增到一个叹为观止的水平，所有通量非常高。<br><br><br><strong>最后是测序信号是怎么得到的？</strong><br>通过碱基上的<strong>荧光基团</strong>。      </p>
<br>

<p><strong>RNA-seq目标：</strong><br>确定特定样品中RNA分子的<strong>序列</strong>、<strong>结构</strong>和<strong>丰度</strong></p>
<ul>
<li>序列指的是A、C、G、U残基的特定顺序；序列可以用于识别一致的蛋白质编码基因、新基因或长的非编码RNA，一旦序列被确定，折叠成二级结构可以揭示分子的类型，如tRAN或miRNA。</li>
<li>结构是指基因结构【即启动子、内含子-外显子街头、5‘端和3’端非翻译区（UTR）级polyA位置】。二级结构提供互补的核苷酸配对和发夹或凸起的位置，三级结构提供分子的三维形状。</li>
<li>丰度指的是每个特定序列的数量值，作为绝对值和归一化的值。</li>
</ul>
<hr>
<h1 id="准备工作"><a href="#准备工作" class="headerlink" title="准备工作"></a>准备工作</h1><p>首先，我们拿到的原始序列文件就是fastq，那么怎么理解fastq文件呢？<br><strong>fastq</strong>和<strong>fasta</strong>这两种sequenced的储存格式是我们后续分析主要处理的两种数据格式。无论是后面的双序列比对（pairwise alignment）还是进化树构建（phylogenetic tree）亦或是高通量分析的回贴（mapping）数据的原始输入都是这两种格式。     </p>
<h2 id="FASTA-amp-FASTQ"><a href="#FASTA-amp-FASTQ" class="headerlink" title="FASTA &amp; FASTQ"></a>FASTA &amp; FASTQ</h2><ul>
<li><strong>FASTA</strong><br>fasta格式是一种非常简单的存储序列的格式，可以储存核酸序列（DNA/RNA）也可以储存蛋白质的氨基酸序列（Amino Acid sequence，简称AA序列）。<br>主要分成2个部分：一个是以“&gt;”为开始的一行主要存储的是序列的描述信息；剩下的是序列部分，中间，前后都可以有空格。序列部分按照官方文档说明应该是小于120，一般是70-80左右。其实实际操作中，程序处理的时候都是自动去掉空格和换行符，把序列读成1行再处理。</li>
<li>举个例子       <figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">&gt;sp|P69905|HBA_HUMAN Hemoglobin subunit alpha OS=Homo sapiens GN=HBA1 </span><br><span class="line">MVLSPADKTNVKAAWGKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLSHGSAQVKGHG</span><br><span class="line">KKVADALTNAVAHVDDMPNALSALSDLHAHKLRVDPVNFKLLSHCLLVTLAAHLPAEFTP</span><br><span class="line">AVHASLDKFLASVSTVLTSKYR </span><br><span class="line">-----</span><br><span class="line">P69905是这个序列在UniRef中的编号，HBA_HUMAN是这个序列的简称，Hemoglobin subunit alpha是全称，&gt;OS=Homo sapiens是物种，GN=HBA1是指基因的名字为HBA1.</span><br><span class="line">&gt;后面的内容就是这个HBA1基因对应的蛋白的序列。</span><br></pre></td></tr></table></figure>


</li>
</ul>
<ul>
<li>再举个例子。</li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">&gt;gi|<span class="number">13650073</span>|gb|AF349571<span class="number">.1</span>| Homo sapiens hemoglobin alpha<span class="number">-1</span> globin chain (HBA1) mRNA, complete cds</span><br><span class="line">CCCACAGACTCAGAGAGAACCCACCATGGTGCTGTCTCCTGACGACAAGACCAACGTCAAGGCCGCCTGG</span><br><span class="line">GGTAAGGTCGGCGCGCACGCTGGCGAGTATGGTGCGGAGGCCCTGGAGAGGATGTTCCTGTCCTTCCCCA</span><br><span class="line">CCACCAAGACCTACTTCCCGCACTTCGACCTGAGCCACGGCTCTGCCCAGGTTAAGGGCCACGGCAAGAA</span><br><span class="line">GGTGGCCGACGCGCTGACCAACGCCGTGGCGCACGTGGACGACATGCCCAACGCGCTGTCCGCCCTGAGC</span><br><span class="line">GACCTGCACGCGCACAAGCTTCGGGTGGACCCGGTCAACTTCAAGCTCCTAAGCCACTGCCTGCTGGTGA</span><br><span class="line">CCCTGGCCGCCCACCTCCCCGCCGAGTTCACCCCTGCGGTGCACGCCTCCCTGGACAAGTTCCTGGCTTC</span><br><span class="line">TGTGAGCACCGTGCTGACCTCCAAATACCGTTAAGCTGGAGCCTCGGTGGCCATGCTTCTTGCCCCTTTG</span><br><span class="line">G</span><br><span class="line">---</span><br><span class="line">对于第<span class="number">1</span>行，所有来自于NCBI的序列都有一个gi号，就是数据库中的流水号，具有唯一性。</span><br><span class="line">gb|AF349571<span class="number">.1</span>是&gt;genebank编号的信息。后面就是对序列的一个通俗的描述。这里使用的是mRNA，包含完整的&gt;CDS（Coding &gt;sequence）区域。</span><br><span class="line">对于下面的序列，大家可能会有一个疑问，为什么mRNA的序列还是用T来表示，而不是U？其实这是为了保证数据&gt;的统一性，因为U只是在RNA中替换了原来的T，所以为了下游的方便分析处理，无论RNA序列还是DNA序列都是使&gt;用T而不是U。</span><br></pre></td></tr></table></figure>



<br>

<ul>
<li><strong>FASTQ</strong></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta">@ST-E00126:128:HJFLHCCXX:2:1101:7405:1133</span></span><br><span class="line">TTGCAAAAAATTTCTCTCATTCTGTAGGTTGCCTGTTCACTCTGATGATAGTTTGTTTTGG</span><br><span class="line">+</span><br><span class="line">FFKKKFKKFKF&lt;KK&lt;F,AFKKKKK7FFK77&lt;FKK,&lt;F7K,,<span class="number">7</span>AF&lt;FF7FKK7AA,<span class="number">7</span>&lt;FA,,</span><br><span class="line">-----</span><br><span class="line">&gt;第<span class="number">1</span>行主要储存序列测序时的坐标等信息</span><br><span class="line"><span class="meta">@ST-E00126:128:HJFLHCCXX:2:1101:7405:1133</span></span><br><span class="line"><span class="meta">@	开始的标记符号			</span></span><br><span class="line">ST-E00126:<span class="number">128</span>:HJFLHCCXX	测序仪唯一的设备名称</span><br><span class="line"><span class="number">2</span>	lane的编号				</span><br><span class="line"><span class="number">1101</span>	tail的坐标</span><br><span class="line"><span class="number">7405</span>	在tail中的X坐标</span><br><span class="line"><span class="number">1133</span>	在tail中的Y坐标</span><br><span class="line">-----</span><br><span class="line">&gt;第<span class="number">2</span>行就是测序得到的序列信息，一般用ATCGN来表示，其中N表示荧光信号干扰无法判断到底是哪个碱基。</span><br><span class="line">&gt;第<span class="number">3</span>行以“+”开始，可以储存一些附加信息，一般是空的。</span><br><span class="line">&gt;第<span class="number">4</span>行储存的是质量信息，与第<span class="number">2</span>行的碱基序列是一一对应的，其中的每一个符号对应的ASCII值成为phred值，可以&gt;简单理解为对应位置碱基的质量值，越大说明测序的质量越好。不同的版本对应的不同。</span><br></pre></td></tr></table></figure>




<h3 id="FASTQ质量值的计算方法"><a href="#FASTQ质量值的计算方法" class="headerlink" title="FASTQ质量值的计算方法"></a>FASTQ质量值的计算方法</h3><p>在测序仪进行测序的时候，会自动根据荧光信号的强弱给出一个参考的测序错误概率（error probility，P）根据定义来说，<strong>P值肯定是越小越好</strong>。我们怎么储存他们呢？直接储存成小数点？比如1%储存成0.01？这肯定是不高效的，因为1个碱基的信息，占用了至少4个字符。<br>所以科学家们的做法想了一个办法：     </p>
<ol>
<li><p>将P取log10之后再乘以-10，得到的结果为Q。<br>  <code>比如，P=1%，那么对应的Q=-10*log10（0.01）=20</code></p>
</li>
<li><p>把这个Q加上33或者64转成一个新的数值，称为Phred，最后把Phred对应的ASCII字符对应到这个碱基。<br>  <code>如Q=20，Phred = 20 + 33 = 53，对应的符号是”5”</code></p>
</li>
</ol>
<p>这样就可以用1个符号与1个碱基一一对应!   </p>
<br>

<h2 id="对测序质量这些东西进行直观化？"><a href="#对测序质量这些东西进行直观化？" class="headerlink" title="对测序质量这些东西进行直观化？"></a>对测序质量这些东西进行直观化？</h2><h3 id="FASTQC的基本介绍"><a href="#FASTQC的基本介绍" class="headerlink" title="FASTQC的基本介绍"></a>FASTQC的基本介绍</h3><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">基本格式</span><br><span class="line">-----</span><br><span class="line">fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] [-c contaminant file] seqfile1 .. seqfileN</span><br><span class="line">-----</span><br><span class="line"><span class="comment">#主要是包括前面的各种选项和最后面的可以加入N个文件</span></span><br><span class="line"><span class="comment">#-o --outdir FastQC生成的报告文件的储存路径，生成的报告的文件名是根据输入来定的</span></span><br><span class="line"><span class="comment">#--extract 生成的报告默认会打包成1个压缩文件，使用这个参数是让程序不打包</span></span><br><span class="line"><span class="comment">#-t --threads 选择程序运行的线程数，每个线程会占用250MB内存，越多越快咯</span></span><br><span class="line"><span class="comment">#-c --contaminants 污染物选项，输入的是一个文件，格式是Name [Tab] Sequence，里面是可能的污染序列，如果&gt;有这个选项，FastQC会在计算时候评估污染的情况，并在统计的时候进行分析，一般用不到</span></span><br><span class="line"><span class="comment">#-a --adapters 也是输入一个文件，文件的格式Name [Tab] Sequence，储存的是测序的adpater序列信息，如果不输入，目前版本的FastQC就按照通用引物来评估序列时候有adapter的残留</span></span><br><span class="line"><span class="comment">#-q --quiet 安静运行模式，一般不选这个选项的时候，程序会实时报告运行的状况。</span></span><br><span class="line">-----</span><br><span class="line">实例：</span><br><span class="line">fastqc -o ./tmp.result/fastQC/ -t <span class="number">6</span> ./tmp.data/fastq/H1EScell-dnase<span class="number">-2014</span>-GSE56869_20151208_SRR1248176_1.fq</span><br></pre></td></tr></table></figure>

<p>运行结束后会生成两个文件：    </p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">H1EScell-dnase<span class="number">-2014</span>-GSE56869_20151208_SRR1248176_1.fq_fastqc.html</span><br><span class="line">H1EScell-dnase<span class="number">-2014</span>-GSE56869_20151208_SRR1248176_1.fq_fastqc.zip</span><br></pre></td></tr></table></figure>





<h3 id="FASTQC的报告介绍"><a href="#FASTQC的报告介绍" class="headerlink" title="FASTQC的报告介绍"></a>FASTQC的报告介绍</h3><ul>
<li>总结信息<br>上图中Summary的部分就是整个报告的目录，整个报告分成若干个部分。合格会有个绿色的对勾，警告是个“!”，不合格是个红色的叉子。</li>
<li>基本信息<br><img src="https://img-blog.csdnimg.cn/20200917211852781.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Encoding指测序平台的版本和相应的编码版本号，这个在计算Phred反推error P的时候有用，如果不明白可以参考之前的文章。</span></span><br><span class="line"><span class="comment">#Total Sequences记录了输入文本的reads的数量</span></span><br><span class="line"><span class="comment">#Sequence length 是测序的长度</span></span><br><span class="line"><span class="comment">#%GC 是我们需要重点关注的一个指标，这个值表示的是整体序列中的GC含量，这个数值一般是物种特意的，比如人类细胞就是42%左右。</span></span><br></pre></td></tr></table></figure>



<ul>
<li>序列测序质量统计<br><img src="https://img-blog.csdnimg.cn/20200917212134860.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#此图中的横轴是测序序列第1个碱基到第101个碱基</span></span><br><span class="line"><span class="comment">#纵轴是质量得分，Q = -10*log10（error P）即20表示1%的错误率，30表示0.1%</span></span><br><span class="line"><span class="comment">#图中每1个boxplot，都是该位置的所有序列的测序质量的一个统计，上面的bar是90%分位数，下面的bar是10%分位数，箱子的中间的横线是50%分位数，箱子的上边是75%分位数，下边是25%分位数</span></span><br><span class="line"><span class="comment">#图中蓝色的细线是各个位置的平均值的连线</span></span><br><span class="line"><span class="comment">#一般要求此图中，所有位置的10%分位数大于20,也就是我们常说的Q20过滤</span></span><br><span class="line"><span class="comment">#所以上面的这个测序结果，需要把后面的87bp以后的序列切除，从而保证后续分析的正确性</span></span><br><span class="line"><span class="comment">#Warning 报警 如果任何碱基质量低于10,或者是任何中位数低于25</span></span><br><span class="line"><span class="comment">#Failure 报错 如果任何碱基质量低于5,或者是任何中位数低于20</span></span><br></pre></td></tr></table></figure>


<ul>
<li>每个tail测序的情况<br><img src="https://img-blog.csdnimg.cn/20200917212446915.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#横轴和之前一样，代表101个碱基的每个不同位置</span></span><br><span class="line"><span class="comment">#纵轴是tail的Index编号</span></span><br><span class="line"><span class="comment">#这个图主要是为了防止，在测序过程中，某些tail受到不可控因素的影响而出现测序质量偏低</span></span><br><span class="line"><span class="comment">#蓝色代表测序质量很高，暖色代表测序质量不高，如果某些tail出现暖色，可以在后续分析中把该tail测序的结果全部都去除</span></span><br></pre></td></tr></table></figure>



<ul>
<li>每条序列的测序质量统计<br><img src="https://img-blog.csdnimg.cn/20200917212618909.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#假如我测的1条序列长度为101bp，那么这101个位置每个位置Q之的平均值就是这条reads的质量值</span></span><br><span class="line"><span class="comment">#该图横轴是0-40，表示Q值</span></span><br><span class="line"><span class="comment">#纵轴是每个值对应的reads数目</span></span><br><span class="line"><span class="comment">#我们的数据中，测序结果主要集中在高分中，证明测序质量良好！</span></span><br></pre></td></tr></table></figure>



<ul>
<li>GC含量统计<br><img src="https://img-blog.csdnimg.cn/20200917212736360.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#横轴是1 - 101 bp；纵轴是百分比</span></span><br><span class="line"><span class="comment">#图中四条线代表A T C G在每个位置平均含量</span></span><br><span class="line"><span class="comment">#理论上来说，A和T应该相等，G和C应该相等，但是一般测序的时候，刚开始测序仪状态不稳定，很可能出现上图的情况。像这种情况，即使测序的得分很高，也需要cut开始部分的序列信息，一般像我碰到这种情况，会cut前面5bp</span></span><br></pre></td></tr></table></figure>



<ul>
<li>序列平均GC含量分布图<br><img src="https://img-blog.csdnimg.cn/20200917212902342.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#横轴是0 - 100%； 纵轴是每条序列GC含量对应的数量</span></span><br><span class="line"><span class="comment">#蓝色的线是程序根据经验分布给出的理论值，红色是真实值，两个应该比较接近才比较好</span></span><br><span class="line"><span class="comment">#当红色的线出现双峰，基本肯定是混入了其他物种的DNA序列</span></span><br><span class="line"><span class="comment">#这张图中的信息良好</span></span><br></pre></td></tr></table></figure>




<ul>
<li>序列测序长度统计<br><img src="https://img-blog.csdnimg.cn/20200917213048760.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#每次测序仪测出来的长度在理论上应该是完全相等的，但是总会有一些偏差</span></span><br><span class="line"><span class="comment">#比如此图中，101bp是主要的，但是还是有少量的100和102bp的长度，不过数量比较少，不影响后续分析</span></span><br><span class="line"><span class="comment">#当测序的长度不同时，如果很严重，则表明测序仪在此次测序过程中产生的数据不可信</span></span><br></pre></td></tr></table></figure>



<ul>
<li>序列Adapter<br><img src="https://img-blog.csdnimg.cn/20200917213134145.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#此图衡量的是序列中两端adapter的情况</span></span><br><span class="line"><span class="comment">#如果在当时fastqc分析的时候-a选项没有内容，则默认使用图例中的四种通用adapter序列进行统计</span></span><br><span class="line"><span class="comment">#本例中adapter都已经去除，如果有adapter序列没有去除干净的情况，在后续分析的时候需要先使用cutadapt软件进行去接头，这个软件以后我会介绍</span></span><br></pre></td></tr></table></figure>



<ul>
<li>重复短序列<br><img src="https://img-blog.csdnimg.cn/20200917213246762.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></li>
</ul>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#这个图统计的是，在序列中某些特征的短序列重复出现的次数</span></span><br><span class="line"><span class="comment">#我们可以看到1-8bp的时候图例中的几种短序列都出现了非常多的次数，一般来说，出现这种情况，要么是adapter没有去除干净，而又没有使用-a参数；要么就是序列本身可能重复度比较高，如建库PCR的时候出现了bias</span></span><br><span class="line"><span class="comment">#对于这种情况，我的办法是可以cut掉前面的一些长度，可以试着cut 5~8bp</span></span><br></pre></td></tr></table></figure>


<p>那么在发现测序数据的质量之后，势必想<strong>剔除低质量的部分</strong>，剔除的软件有许多，譬如：<strong>Trimmomatic</strong> ， <strong>fastp</strong> ，<strong>cutadapter</strong>。<br>一般给定的标准是<strong>清除存在的所有接头序列</strong>，<strong>过滤掉q小于20的碱基</strong>，<strong>去除N碱基大于5%的序列</strong>，<strong>去除A与T或者C与G含量相差10%的序列</strong>，这个标准一般通用，具体可以根据自己数据去筛选。</p>
<br>   

<h2 id="质控——去接头、去污染、去重复"><a href="#质控——去接头、去污染、去重复" class="headerlink" title="质控——去接头、去污染、去重复"></a>质控——去接头、去污染、去重复</h2><h3 id="fastx-Toolkit"><a href="#fastx-Toolkit" class="headerlink" title="fastx Toolkit"></a>fastx Toolkit</h3><p>fastx Toolkit是包含处理fastq/fasta文件的一系列的工具，它是基于java开发的，我们高通量测序最常用到的是使用这个软件进行reads的裁剪（trim）。<br>它的安装可以根据官网的指南进行：<code>http://hannonlab.cshl.edu/fastx_toolkit/index.html</code></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line">FASTQ-to-FASTA</span><br><span class="line">说明：这个命令主要是用来转换FASTA格式与FASTQ格式$ fastq_to_fasta [-h] [-r] [-n] [-v] [-z] [-i] [-o]</span><br><span class="line">[-h] = 获得帮助信息.</span><br><span class="line">[-r] = 使用序号去代替fastq文件中原来的reads名.</span><br><span class="line">[-n] = 如果fastq中有N，保留.（默认是有N的序列删除）</span><br><span class="line">[-v] = 报告reads的总数</span><br><span class="line">[-z] = 调用GZip软件，输出的文件自动经过压缩.</span><br><span class="line">[-i] = 输入文件，可以是fastq/fasta格式.</span><br><span class="line">[-o] =输出路径，如果不设置会直接输出到屏幕.</span><br><span class="line"></span><br><span class="line">FASTX Statistics</span><br><span class="line">说明：主要统计一下序列的基本信息，如GC含量什么的，很少用，基本使用FastQC代替</span><br><span class="line">$ fastx_quality_stats </span><br><span class="line">[-h] [-i INFILE] [-o OUTFILE]</span><br><span class="line">[-h] = 获得帮助信息</span><br><span class="line">[-i] = FASTA/Q格式的输入文件</span><br><span class="line">[-o] = 输出路径，如果不设置会直接输出到屏幕.</span><br><span class="line"></span><br><span class="line">FASTA/Q Clipper</span><br><span class="line">说明：主要是进行reads的过滤和adapter的裁剪</span><br><span class="line">$ fastx_clipper [-h] [-a ADAPTER] [-i INFILE] [-o OUTFILE]</span><br><span class="line"><span class="comment">#参数很多，我们慢慢来看！</span></span><br><span class="line">[-h] = 获得帮助信息.</span><br><span class="line">[-a ADAPTER] = Adapter序列信息. 默认的是CCTTAAGG</span><br><span class="line">[-l N] = 如果<span class="number">1</span>条reads小于N就抛弃，默认<span class="number">5.</span></span><br><span class="line">[-d N] = 保留adapter并保留后面的Nbp，如果设置-d <span class="number">0</span>等于没有用这个参数.</span><br><span class="line">[-c] = 只保留包含adapter的序列</span><br><span class="line">[-C] = 只保留不包含adapter的序列</span><br><span class="line">[-k] = 报告adater的序列信息</span><br><span class="line">[-n] = 如果reads中有N，保留reads.（默认是有N的序列删除）</span><br><span class="line">[-v] = 报告序列总数</span><br><span class="line">[-z] = 调用GZip软件，输出的文件自动经过压缩.</span><br><span class="line">[-D]= Debug output.</span><br><span class="line"></span><br><span class="line">FASTA/Q Trimmer</span><br><span class="line">说明：这个是我最常用的工具，可以快速切序列</span><br><span class="line">$ fastx_trimmer [-h] [-f N] [-l N] [-z] [-v] [-i INFILE] [-o OUTFILE]</span><br><span class="line">[-h] = 获得帮助信息.</span><br><span class="line">[-f N] = 序列中从第几个碱基开始保留. 默认是<span class="number">1.</span></span><br><span class="line">[-l N] = 序列最后保留到多少个碱基，默认是整条序列全部保留.</span><br><span class="line">[-z] = 调用GZip软件，输出的文件自动经过压缩.</span><br><span class="line"></span><br><span class="line">备注：其实Fastx工具箱中还有很多其他的工具，但是都不是特别常用，所以我也就不进行介绍了。如果有感兴趣的同学，请直接到他们的官网上去阅读说明文档。</span><br></pre></td></tr></table></figure>

<br>

<h3 id="cutadapt"><a href="#cutadapt" class="headerlink" title="cutadapt"></a>cutadapt</h3><p>cutadapt软件是最常用的去adapter的工具。它是基于Python编写的一个Python包.<br>cutadapt程序包的下载地址 : <code>https://pypi.python.org/pypi/cutadapt</code>   </p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br></pre></td><td class="code"><pre><span class="line">一些最基本的用法</span><br><span class="line"><span class="comment"># cutadapt的功能特别强大，相对应的参数真的特别说，有几十个参数，我们平时只会用到很少的几个，我在这里为大家介绍一下。</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 最基本的形式，可以去掉3‘端的adapter序列</span></span><br><span class="line">$ cutadapt -a AACCGGTT -o output.fastq input.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment"># 可以直接输入或者输出压缩文件，不需要修改参数，输出文件的后面加上.gz</span></span><br><span class="line">$ cutadapt -a AACCGGTT -o output.fastq.gz input.fastq.gz</span><br><span class="line"></span><br><span class="line"><span class="comment"># 假如去掉3‘端的adapter AAAAAAA 和5’端的adapter TTTTTTT</span></span><br><span class="line">$ cutadapt -a AAAAAAA -g TTTTTTT -o output.fastq input.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment"># cutadapt也可以用来进行reads的cut</span></span><br><span class="line"><span class="comment"># 去掉最前面的5bp</span></span><br><span class="line">$ cutadapt -u <span class="number">5</span> -o trimmed.fastq input_reads.fastq</span><br><span class="line"></span><br><span class="line"><span class="comment"># 进行reads测序质量的过滤</span></span><br><span class="line"><span class="comment"># cutadapt软件可以使用-q参数进行reads质量的过滤。基本原理就是，一般reads头和尾会因为测序仪状态或者是反应时间的问题造成测序质量差，比较粗略的一个过滤办法就是-q进行过滤。需要特别说明的是，这里的-q对应的数字和phred值是不一样的，它是软件根据一定的算法计算出来的。</span></span><br><span class="line"><span class="comment"># 3‘端进行一个简单的过滤,--quality-base=33是指序列使用的是phred33计分系统</span></span><br><span class="line">$ cutadapt -q <span class="number">10</span> --quality-base=<span class="number">33</span> -o output.fastq input.fastq </span><br><span class="line"><span class="comment"># 3‘端 5’端都进行过滤,3'的阈值是10，5‘的阈值是15</span></span><br><span class="line">$ cutadapt -q <span class="number">10</span>,<span class="number">15</span> --quality-base=<span class="number">33</span> -o output.fastq input.fastq </span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Reads 长度的过滤</span><br><span class="line">[--minimum-length N <span class="keyword">or</span> -m N]</span><br><span class="line"><span class="comment"># 当序列长度小于N的时候，reads扔掉</span></span><br><span class="line"></span><br><span class="line">[--too-short-output FILE]</span><br><span class="line"><span class="comment"># 上面参数获得的这些序列不是直接扔掉，而是输出到一个文件中</span></span><br><span class="line"></span><br><span class="line">[--maximum-length N <span class="keyword">or</span> -M N]</span><br><span class="line"><span class="comment"># 当序列长度大于N的时候，reads扔掉</span></span><br><span class="line"></span><br><span class="line">[--too-long-output FILE]</span><br><span class="line"><span class="comment"># 上面参数获得的这些序列不是直接扔掉，而是输出到一个文件中</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Paired-Reads的裁剪（trim）</span><br><span class="line"><span class="comment"># 现在很多的测序都是双端测序，那么从测序原理上来说，一对reads来自于1簇反应，所以一起进行adapter的trim可能效果更好。cutadapt自然也提供了这样的功能。</span></span><br><span class="line">$ cutadapt -a ADAPTER_FWD -A ADAPTER_REV -o out<span class="number">.1</span>.fastq -p out<span class="number">.2</span>.fastq reads<span class="number">.1</span>.fastq reads<span class="number">.2</span>.fastq</span><br><span class="line"><span class="comment"># -a是第1个文件的adapter序列</span></span><br><span class="line"><span class="comment"># -A是第2个文件的adapter序列</span></span><br><span class="line"><span class="comment"># -o是第1个输出文件</span></span><br><span class="line"><span class="comment"># -p是第2个输出文件</span></span><br></pre></td></tr></table></figure>

<br>

<h3 id="Trimmomatic"><a href="#Trimmomatic" class="headerlink" title="Trimmomatic"></a>Trimmomatic</h3><p>是一个功能强大的数据过滤软件。主要用来去除 Illumina 平台的 Fastq 序列中的接头，并根据碱基质量值对 Fastq 进行修剪。<br>软件有两种过滤模式，分别对应 SE 和 PE 测序数据，同时支持 gzip 和 bzip2 压缩文件。<br>另外也支持 phred-33 和 phred-64 格式互相转化，不过现在绝大部分 Illumina 平台的产出数据也都转为使用 phred-33 格式了。</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#基本格式Usage: </span></span><br><span class="line">       trimmomatic PE [-version] [-threads &lt;threads&gt;] [-phred33|-phred64] [-trimlog &lt;trimLogFile&gt;] [-summary &lt;statsSummaryFile&gt;] [-quiet] [-validatePairs] [-basein &lt;inputBase&gt; | &lt;inputFile1&gt; &lt;inputFile2&gt;] [-baseout &lt;outputBase&gt; | &lt;outputFile1P&gt; &lt;outputFile1U&gt; &lt;outputFile2P&gt; &lt;outputFile2U&gt;] &lt;trimmer1&gt;...</span><br><span class="line">   <span class="keyword">or</span>: </span><br><span class="line">       trimmomatic SE [-version] [-threads &lt;threads&gt;] [-phred33|-phred64] [-trimlog &lt;trimLogFile&gt;] [-summary &lt;statsSummaryFile&gt;] [-quiet] &lt;inputFile&gt; &lt;outputFile&gt; &lt;trimmer1&gt;...</span><br><span class="line"><span class="comment">#参数说明</span></span><br><span class="line"><span class="comment">#PE/SE             设定对Paired-End或Single-End的reads进行处理，其输入和输出参数稍有不一样。</span></span><br><span class="line"><span class="comment">#-threads          设置多线程运行数</span></span><br><span class="line"><span class="comment">#-phred33          设置碱基的质量格式，可选pred64，不设置这个参数，软件会自动判断输入文件是哪种格式</span></span><br><span class="line"><span class="comment">#-trimlog          指定过滤日志文件名，日志中包含以下四列内容:read ID、过滤之后剩余序列长度、过滤之后的序列起始碱基位置（序列开头处被切掉了多少个碱基）、过滤之后的序列末端碱基位置、序列末端处被剪切掉的碱基数。由于生成的trimlog文件中包含了每一条 reads 的处理记录，因此文件体积巨大（GB级别），如果后面不会用到 trim日志，建议不要使用这个参数</span></span><br><span class="line"><span class="comment">#-basein           通常 PE 测序的两个文件，指定其中 R1 文件名，软件会推测出 R2 的文件名，但是这个功能实测并不好用，建议不用-basein参数，直接指定两个文件名（R1 和 R2）作为输入</span></span><br><span class="line"><span class="comment">#-baseout          输出文件有四个，使用 -baseout 参数指定输出文件的 basename，软件会自动为四个输出文件命名，过滤之后双端序列都保留的就是 paired，反之如果其中一端序列过滤之后被丢弃了另一端序列保留下来了就是 unpaired(即 成对的clean reads, 未成对的正向序列以及未成对的反向序列)一般情况下，若paired reads百分比占90%以上，可只对paired reads进行比对分析ILLUMINACLIP:TruSeq3-PE.fa:2:30:10:8:true切除adapter序列。参数后面分别接adapter序列的fasta文件：第一步 seed 搜索时允许的最大错配碱基个数2：palindrome模式下匹配碱基数阈值30：simple模式下的匹配碱基数阈值10(7-15之间）：palindrome 模式允许切除的最短接头序列为 8bp（默认值）：palindrome 模式去除与 R1 完全反向互补的 R2（默认去除false），但在有些情况下，例如需要用到 paired reads 的 bowtie2 流程，就要将这个参数改为 true，否则会损失一部分 paired reads。按照规定顺序，ILLUMINACLIP 各个参数之间以冒号分开，PE测序需要注意最后一个参数。对于SE测序最后两个参数可以不设置</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#LEADING:3         切除首端碱基质量小于3的碱基,Illumina平台有些低质量的碱基质量值被标记为 2 ，所以设置为3可以过滤掉这部分低质量碱基。</span></span><br><span class="line"><span class="comment">#TRAILING:3        切除尾端碱基质量小于3的碱基</span></span><br><span class="line"><span class="comment">#SLIDINGWINDOW:15:20 滑窗质量过滤，一般一个read的低质量序列都是集中在末端，也有很少部分在开头。从5'端开始进行滑动，当滑动位点周围一段序列(window)的平均碱基低于阈值，则从该处进行切除。Windows的size是15个碱基（一般设置在10-30之间），其平均碱基质量小于20，则切除</span></span><br><span class="line"><span class="comment">#MINLEN:50         可被保留的最短reads长度，应根据原始序列的长度而定</span></span><br><span class="line"><span class="comment">#HEADCROP:&lt;length&gt; 在reads的首端切除指定的长度</span></span><br><span class="line"><span class="comment">#CROP:&lt;length&gt;     保留reads到指定的长度</span></span><br><span class="line"><span class="comment">#TOPHRED33         将碱基质量转换为pred33格式</span></span><br><span class="line"><span class="comment">#TOPHRED64         将碱基质量转换为pred64格式</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#示例参考：</span></span><br><span class="line">trimmomatic PE -phred33 /data/hushy/seqdata/RESP/study11/raw_reads/SRR6994256_1.fastq.gz /data/hushy/seqdata/RESP/study11/raw_reads/SRR6994256_2.fastq.gz -baseout /data/hushy/seqdata/RESP/study11/analysis/SRR6994256.fastq.gz ILLUMINACLIP:/export/home/hushy/miniconda2/pkgs/trimmomatic<span class="number">-0.39</span><span class="number">-1</span>/share/trimmomatic<span class="number">-0.39</span><span class="number">-1</span>/adapters/TruSeq3-PE.fa:<span class="number">2</span>:<span class="number">30</span>:<span class="number">10</span> LEADING:<span class="number">3</span> TRAILING:<span class="number">3</span> SLIDINGWINDOW:<span class="number">4</span>:<span class="number">15</span> MINLEN:<span class="number">50</span></span><br><span class="line"><span class="comment"># /export/home/hushy/miniconda2/pkgs/trimmomatic-0.39-1/share/trimmomatic-0.39-1/adapters为接头序列的储存路径，其中TruSeq3-PE.fa与TruSeq3-SE.fa为illumina hiseq和miseq的接头，如果不是使用hiseq2000的，到illumina官网下载接头序列并放到这个文件夹中。TruSeq2对应着illumina GAII测序。</span></span><br><span class="line"><span class="comment"># 默认输出文件格式: SRR6994256_1P.fastq.gz  SRR6994256_1U.fastq.gz  SRR6994256_2P.fastq.gz SRR6994256_2U.fastq.gz</span></span><br><span class="line"></span><br><span class="line">trimmomatic PE -phred33 ~/SRR733/SRR2854733_1.fastq ~/SRR733/SRR2854733_2.fastq \</span><br><span class="line">   ~/SRR733/clsseq/SRR2854733_1_paired.fq ~/SRR733/clsseq/SRR2854733_1_unpaired.fq \</span><br><span class="line">   ~/SRR733/clsseq/SRR2854733_2_paired.fq ~/SRR733/clsseq/SRR2854733_2_unpaired.fq \</span><br><span class="line">   ILLUMINACLIP:/usr/local/src/Trimmomatic/Trimmomatic<span class="number">-0.36</span>/adapters/TruSeq3-PE.fa:<span class="number">2</span>:<span class="number">30</span>:<span class="number">10</span> \</span><br><span class="line">   LEADING:<span class="number">3</span> TRAILING:<span class="number">3</span> SLIDINGWINDOW:<span class="number">10</span>:<span class="number">15</span> HEADCROP:<span class="number">8</span> MINLEN:<span class="number">36</span></span><br></pre></td></tr></table></figure>

<br>   

<h3 id="Trim-Galore"><a href="#Trim-Galore" class="headerlink" title="Trim Galore"></a>Trim Galore</h3><p>Trim Galore是对FastQC和Cutadapt的包装。适用于所有高通量测序，包括RRBS(Reduced Representation Bisulfite-Seq ), Illumina、Nextera 和smallRNA测序平台的双端和单端数据。</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#示例参考：</span></span><br><span class="line">trim_galore -q <span class="number">20</span> --phred33 --stringency <span class="number">3</span> --length <span class="number">20</span> -e <span class="number">0.1</span> </span><br><span class="line">            --paired $dir/cmp/<span class="number">01</span>raw_data/$fq1 $dir/cmp/<span class="number">01</span>raw_data/$fq2   </span><br><span class="line">            --gzip -o $input_data</span><br><span class="line"></span><br><span class="line"><span class="comment">#参数说明</span></span><br><span class="line"><span class="comment">#--quality：设定Phred quality score阈值，默认为20。</span></span><br><span class="line"><span class="comment">#--phred33：：选择-phred33或者-phred64，表示测序平台使用的Phred quality score。</span></span><br><span class="line"><span class="comment">#--adapter：输入adapter序列。也可以不输入，Trim Galore!会自动寻找可能性最高的平台对应的adapter。自动搜选的平台三个，也直接显式输入这三种平台，即--illumina、--nextera和--small_rna。</span></span><br><span class="line"><span class="comment">#--stringency：设定可以忍受的前后adapter重叠的碱基数，默认为1（非常苛刻）。可以适度放宽，因为后一个adapter几乎不可能被测序仪读到。</span></span><br><span class="line"><span class="comment">#--length：设定输出reads长度阈值，小于设定值会被抛弃。</span></span><br><span class="line"><span class="comment">#--paired：对于双端测序结果，一对reads中，如果有一个被剔除，那么另一个会被同样抛弃，而不管是否达到标准。</span></span><br><span class="line"><span class="comment">#--retain_unpaired：对于双端测序结果，一对reads中，如果一个read达到标准，但是对应的另一个要被抛弃，达到标准的read会被单独保存为一个文件。</span></span><br><span class="line"><span class="comment">#--gzip和--dont_gzip：清洗后的数据zip打包或者不打包。</span></span><br><span class="line"><span class="comment">#--output_dir：输入目录。需要提前建立目录，否则运行会报错。</span></span><br><span class="line"><span class="comment">#-- trim-n : 移除read一端的reads</span></span><br></pre></td></tr></table></figure>

<h3 id="举个例子"><a href="#举个例子" class="headerlink" title="举个例子"></a>举个例子</h3><p>其实在实际中，我们从公司拿到的数据大多数已经进行过cutadapt，我们其实更关注的是reads的trim。我们利用我们的测试数据进行一下实际处理说明。</p>
<p>我们首先先要用fastqc对test1.fastq与test2.fastq进行一下质量评估，评估的主要结果如下：</p>
<p>read1的测序质量图<br><img src="https://img-blog.csdnimg.cn/20200918095138875.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="read1的测序质量图"><br>read2的测序质量图：<br><img src="https://img-blog.csdnimg.cn/20200918095240207.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></p>
<p>我们从上面两张图可以明显看出，read1的测序质量明显好于read2，一般我们确定要trim多少bp的时候都是按phred20这个标准进行评估。比如，对于我们测试数据，read1就不需要trim，read2需要保留1-85bp。对应的fastx_trimmer的命令如下：<br><code>fastx_trimmer -i test_data_2.fastq -o test_data_2_trim.fastq -f 1 -l 85</code>   </p>
<br>

<p><strong>至此，我们就得到了一份高质量的原始数据（clean data），从而正式进入分析工作</strong>    </p>
<hr>
<h1 id="上游分析"><a href="#上游分析" class="headerlink" title="上游分析"></a>上游分析</h1><p>无论是以前的bowtie2+samtools+cufflinks+deseq2，还是现在转录组的当红炸子鸡流程     hisat2+stringtie+ballgown，其本质的工作流程其实是一样的，只不过使用的算法不同而已。   </p>
<h2 id="比对回帖"><a href="#比对回帖" class="headerlink" title="比对回帖"></a>比对回帖</h2><h3 id="这一步是干什么的？"><a href="#这一步是干什么的？" class="headerlink" title="这一步是干什么的？"></a>这一步是干什么的？</h3><p>首先，我们的fastq文件存储的数据是一个零散的状态，那要怎么样把它恢复到打断前的状态呢？<br>这里我们需要一个模板，按照模板，把序列排序：    </p>
<p><img src="https://img-blog.csdnimg.cn/20200918100400878.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"><br>这里的<strong>ref就是模板</strong>，即<strong>参考基因组</strong>，而我们的<strong>fastq文件本质就是一条一条的小序列</strong>，<strong>在模板的指引下，我们得到了他们原本在基因组上应该在位置</strong>，这一步就是回帖的含义。      </p>
<p>也即是bowtie2跟hisat2所干的事    </p>
<ul>
<li>比对到参考基因组（genome）<br>1、将reads作图到基因组提供基因组的位置信息，这可以用于<strong>发现新的基因和转录本</strong>，以及用于<strong>量化表达</strong>。<br>2、面临困难：reads相对较短，数目以百万计，而基因组可能很大，并且包含非唯一的序列，如<strong>重复和假基因</strong>，降低这些区域的可作图性；比对程序必须应对由<strong>基因组变异和测序错误造成的不匹配和插入/缺失</strong>；许多生物在它们的基因中<strong>有内含子 ，因此RNA-seq读段非连续地比对到基因组</strong>。</li>
</ul>
<p>RNA-seq读段专化的剪接比对程序（spliced aligner）使用不同的方法来比对剪接的读段，包括执行一个初始化的比对，以发现外显子连接（exon junction），然后引导最终的比对。如果有基因组注释可用，比对程序可以使用它来放置剪接的读段。</p>
<p><strong>当为RNA-seq研究选择一个比对程序时，主要考虑是否需要剪接比对（spliced alignment）。</strong></p>
<ul>
<li>如果生物体没有内含子，或对小分子RNA（microRNA）进行测序，完全可以使用连续地比对程序，如Bowtie或BWA，它们最初是为DNA开发的。</li>
<li>如果读段被作图到一个<strong>转录组</strong>而不是一个基因组，也可以使用这些比对程序。</li>
<li>如果RNA-seq读段被作图到*<em>含有内含子的基因组</em>，则必须使用剪接比对程序，如TopHat、STAR或GSNAP。（多数情况）</li>
</ul>
<p>由于二代测序的reads长度通常介于50~300个碱基，因此<strong>即便使用双端测序，也基本不可能覆盖完整的mRNA转录本</strong>，因此想直接用FastQ文件从头分析测到了哪些转录本需要非常复杂的分析和计算。好在通常情况下，<strong>公共数据库已经提供了测序样品的基因组和转录本的序列</strong>。因此我们只需要知道，每一条reads来自哪一条转录本就可以了，<strong>这个将reads与参考（Reference）基因组/转录组的序列进行比较和匹配的过程，我们通常称之为“比对”</strong>（文献中提到的read alignment和mapping通常说的都是这个）。</p>
<br>

<p>转录组测序的比对通常分为基因组比对和转录组比对两种，顾名思义，基因组比对就是把reads比对到完整的基因组序列上，而转录组比对则是把reads比对到所有已知的转录本序列上。如果不是很急或者只想知道已知转录本表达量，个人建议使用基因组比对的方法进行分析，理由如下：</p>
<ol>
<li><strong>转录组比对需要准确的已知转录本的序列</strong>，对于来自未知转录本（比如一些未被数据库收录的lncRNA）或序列不准确的reads无法正确比对；</li>
<li>与上一条类似，<strong>转录组比对不能对转录本的可变剪接进行分析，数据库中未收录的剪接位点会被直接丢弃</strong>；</li>
<li>由于<strong>同一个基因存在不同的转录本，因此很多reads可以同时完美比对到多个转录本，reads的比对评分会偏低，可能被后续计算表达量的软件舍弃，影响后续分析</strong>（有部分软件解决了这个问题）；</li>
<li>由于与DNA测序使用的参考序列不同，因此不利于RNA和DNA数据的整合分析。</li>
</ol>
<p><strong>而上面的问题使用基因组比对都可以解决。</strong></p>
<br>

<p>此外，值得注意的是，<strong>RNA测序并不能直接使用DNA测序常用的BWA、Bowtie等比对软件，这是由于真核生物内含子的存在，导致测到的reads并不与基因组序列完全一致</strong>，<strong>因此需要使用Tophat/HISAT/STAR等专门为RNA测序设计的软件进行比对</strong>。</p>
<p><strong>HISAT2</strong>，取代Bowtie/TopHat程序，能够<strong>将RNA-Seq的读取与基因组</strong>进行快速比对。HISAT利用大量<strong>FM索引</strong>，以覆盖整个基因组。Index的目的主要使用与序列比对。<strong>由于物种的基因组序列比较长， 如果将测序序列与整个基因组进行比对，则会非常耗时</strong>。因此采用将测序序列和参考基因组的Index文件进行比对，会节省很多时间。以人类基因组为例，它需要48,000个索引，每个索引代表~64,000 bp的基因组区域。这些小的索引结合几种比对策略，实现了RNA-Seq读取的高效比对，特别是那些跨越多个外显子的读取。尽管它利用大量索引，但HISAT只需要4.3 GB的内存。这种应用程序支持任何规模的基因组，包括那些超过40亿个碱基的。</p>
<h3 id="Hisat2"><a href="#Hisat2" class="headerlink" title="Hisat2"></a>Hisat2</h3><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br></pre></td><td class="code"><pre><span class="line">构建索引：</span><br><span class="line">hisat2-build [options]* &lt;reference_in&gt; &lt;ht2_base&gt;</span><br><span class="line"></span><br><span class="line"><span class="comment">#&lt;reference_in&gt; ：fasta文件 list，如果为list，使用逗号分开</span></span><br><span class="line"><span class="comment">#&lt;ht2_base&gt; ：索引文件的前缀名，如设为xxx，则生成的索引文件为xxx.1.ht2,xxx.2.ht2，默认的前缀名为NAME</span></span><br><span class="line"><span class="comment">#option:详见说明书</span></span><br><span class="line"></span><br><span class="line">比对：</span><br><span class="line">hisat2 [options]* -x &lt;hisat2-idx&gt; &#123;<span class="number">-1</span> &lt;m1&gt; <span class="number">-2</span> &lt;m2&gt; | -U &lt;r&gt; | --sra-acc &lt;SRA accession number&gt;&#125; [-S &lt;hit&gt;]</span><br><span class="line"><span class="comment">#参数说明</span></span><br><span class="line"><span class="comment">#-p ：线程数目</span></span><br><span class="line"><span class="comment">#--dta  ：注意！！！在下游使用stringtie组装的时候一定要在hisat中设置这个参数！！！</span></span><br><span class="line"><span class="comment">#-x &lt;hisat2-idx&gt; ：参考基因组索引的basename，即前缀名</span></span><br><span class="line"><span class="comment">#&#123;&#125;：其中的内容意思为hisat2可以接受单端测序，双端测序，或者直接提交SRA ID号</span></span><br><span class="line"><span class="comment">#-1 &lt;m1&gt; ：双端测序的read1 list ，若为list，使用逗号隔开，名字与2要匹配，如-1 flyA_1.fq,flyB_1.fq</span></span><br><span class="line"><span class="comment">#-2 &lt;m2&gt; ：双端测序的read2 list ，若为list，使用逗号隔开，名字与1要匹配，如-2 flyA_2.fq,flyB_2.fq</span></span><br><span class="line"><span class="comment">#-U &lt;r&gt;：单端测序list，若为list，使用逗号隔开，-U lane1.fq,lane2.fq,lane3.fq,lane4.fq</span></span><br><span class="line"><span class="comment">#--sra-acc &lt;SRA accession number&gt; : SRAID list，若为list，使用逗号隔开，--sra-acc SRR353653,SRR353654</span></span><br><span class="line"><span class="comment">#-S &lt;hit&gt; ：SAM写入的文件名，默认写入到标准输出中</span></span><br><span class="line"><span class="comment">#options:这里只列出可调节的类别，至于参数调整，详见说明书</span></span><br><span class="line"><span class="comment">#Input options</span></span><br><span class="line"><span class="comment">#Alignment options</span></span><br><span class="line"><span class="comment">#Scoring options</span></span><br><span class="line"><span class="comment">#Spliced alignment options（重要）</span></span><br><span class="line"><span class="comment">#Reporting options</span></span><br><span class="line"><span class="comment">#Paired-end options（重要）</span></span><br><span class="line"><span class="comment">#Output options（重要）</span></span><br><span class="line"><span class="comment">#SAM options</span></span><br><span class="line"><span class="comment">#Performance options</span></span><br><span class="line"><span class="comment">#Other options</span></span><br></pre></td></tr></table></figure>

<h4 id="举个例子-1"><a href="#举个例子-1" class="headerlink" title="举个例子"></a>举个例子</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br></pre></td><td class="code"><pre><span class="line">hisat2-build human.fa --snp human.snp human_hg38</span><br><span class="line"><span class="comment">#最终会生成以human_hg38为前缀，以.*.ht2为后缀的索引文件</span></span><br><span class="line"><span class="comment">#其中提供了snp信息</span></span><br><span class="line"></span><br><span class="line">单端测序比对</span><br><span class="line">hisat2 -f -x human_hg38 -U reads_1.fa -S eg1.sam</span><br><span class="line"><span class="comment">#-f:输入文件为fasta文件</span></span><br><span class="line"><span class="comment">#-q：输入文件为fastq文件</span></span><br><span class="line"><span class="comment">#-x：只写前缀</span></span><br><span class="line"><span class="comment">#最终生成eg1.sam文件</span></span><br><span class="line"></span><br><span class="line">双端测序比对</span><br><span class="line">hisat2 -f -x human_hg38 <span class="number">-1</span> reads_1.fa <span class="number">-2</span> reads_2.fa -S eg2.sam</span><br><span class="line"><span class="comment">#最后生成eg2.sam</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">可以使用samtools或者bcftools作为下游</span><br><span class="line">samtools view -bS eg2.sam &gt; eg2.bam</span><br><span class="line"><span class="comment">#转化为bam文件</span></span><br><span class="line">samtools sort eg2.bam -o eg2.sorted.bam</span><br><span class="line"><span class="comment">#bam文件转变为sorted bam</span></span><br><span class="line">samtools mpileup -uf human.fa eg2.sorted.bam | bcftools view -bvcg - &gt; eg2.raw.bcf</span><br><span class="line"><span class="comment">#生成bcf文件</span></span><br><span class="line">bcftools view eg2.raw.bcf</span><br><span class="line"><span class="comment">#查看bcf文件</span></span><br><span class="line"><span class="comment">#详见samtools工具的使用</span></span><br></pre></td></tr></table></figure>

<h3 id="Bowtie2"><a href="#Bowtie2" class="headerlink" title="Bowtie2"></a>Bowtie2</h3><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br></pre></td><td class="code"><pre><span class="line">创建索引</span><br><span class="line">bowtie2-build /data/database/kraken2/NT/library/nt/library.fna index</span><br><span class="line"></span><br><span class="line">基本用法</span><br><span class="line">bowtie2 [options]* -x &lt;index&gt; &#123;<span class="number">-1</span> &lt;m1&gt; <span class="number">-2</span> &lt;m2&gt; | -U &lt;r&gt; | --interleaved &lt;i&gt; | -b &lt;bam&gt;&#125; [-S &lt;sam&gt;]</span><br><span class="line"></span><br><span class="line">必选参数</span><br><span class="line"><span class="comment">#-x  由bowtie2-build所生成的索引文件的前缀,需要指定路径及其共用文件名</span></span><br><span class="line"><span class="comment">#-1  使用trimmomatic质控后与read2配对（paired）的read1。可以为多个文件，并用逗号分开；多个文件必须和 -2 &lt;m2&gt; 中制定的文件一一对应。</span></span><br><span class="line"><span class="comment">#-2  使用trimmomatic质控后与read1配对的read2</span></span><br><span class="line"><span class="comment">#-U  使用trimmomatic质控后未配对（unpaired）的reads。可以为多个文件，并用逗号分开，测序文件中的reads的长度可以不一样。</span></span><br><span class="line"><span class="comment">#-S  所生成的SAM格式的文件前缀。默认是输入到标准输出。</span></span><br><span class="line"></span><br><span class="line">可选参数</span><br><span class="line"><span class="number">1.</span>输入参数</span><br><span class="line"><span class="comment">#-q                输入的文件为FASTQ格式文件，此项为默认值</span></span><br><span class="line"><span class="comment">#-f                输入的文件为FASTA格式文件</span></span><br><span class="line"><span class="comment">#-5/--trim5 &lt;int&gt;  剪掉5'端&lt;int&gt;长度的碱基，再用于比对。(default: 0)</span></span><br><span class="line"><span class="comment">#-3/--trim3 &lt;int&gt;  剪掉3'端&lt;int&gt;长度的碱基，再用于比对。(default: 0).</span></span><br><span class="line"><span class="comment">#--phred33         输入的碱基质量等于ASCII+33</span></span><br><span class="line"></span><br><span class="line"><span class="number">2.</span>Paired-end 参数</span><br><span class="line"><span class="comment">#--no-mixed       默认设置下, 一对reads不能成对比对到参考序列上, 则单独对每个read进行比对. 该选项则阻止此行为.</span></span><br><span class="line"><span class="comment">#--no-discordant  默认设置下, 一对reads不能和谐比对(concordant alignment,即满足-I, -X, --fr/--rf/--ff的条件)到参考序列上, 则搜寻其不和谐比对(disconcordant alignment, 即两条reads都能独一无二地比对到参考序列上, 但是不满足-I,-X,--fr/--rf/--ff的条件). 该选项阻止此行为.</span></span><br><span class="line"></span><br><span class="line"><span class="number">3.</span>–end-to-end模式下的预设参数</span><br><span class="line"><span class="comment">#--end-to-end 比对是将整个read和参考序列进行比对. 该模式--ma的值为0. 该模式为默认模式, --local模式冲突.</span></span><br><span class="line"><span class="comment">#--local 该模式下对read进行局部比对, 从而, read两端的一些碱基不比对，从而使比对得分满足要求. 该模式下 –ma默认为2.</span></span><br><span class="line"><span class="comment">#--very-fast Same as: -D 5 -R 1 -N 0 -L 22 -i S,0,2.50 </span></span><br><span class="line"><span class="comment">#--fast Same as: -D 10 -R 2 -N 0 -L 22 -i S,0,2.50 </span></span><br><span class="line"><span class="comment">#--sensitive Same as: -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default in --end-to-end mode) </span></span><br><span class="line"><span class="comment">#--very-sensitive Same as: -D 20 -R 3 -N 0 -L 20 -i S,1,0.50</span></span><br><span class="line"></span><br><span class="line"><span class="number">4.</span>报告参数</span><br><span class="line"><span class="comment">#-k   默认设置下, bowtie2搜索出了一个read不同的比对结果, 并报告其中最好的比对结果(如果好几个最好的比对结果得分一致, 则随机挑选出其中一个). 而在该模式下, bowtie2最多搜索出一个read &lt;int&gt;个比对结果, 并将这些结果按得分降序报告出来.</span></span><br><span class="line"><span class="comment">#-a   和-k参数一样, 不过不限制搜索的结果数目. 并将所有的比对结果都按降序报告出来. 此参数和-k参数冲突. 值得注意的是: 如果基因组含有很多重复序列时, 该参数会导致程序运行极其缓慢.</span></span><br><span class="line"></span><br><span class="line"><span class="number">5.</span>输出参数</span><br><span class="line"><span class="comment">#-t/--time  --un &lt;path&gt;        将unpaired reads写入到&lt;path&gt;.</span></span><br><span class="line"><span class="comment">#--no-unal                     不能map到GENOME的reads,不保留sam记录</span></span><br><span class="line"><span class="comment">#--un-conc &lt;path&gt;              不能map到GENOME的reads，fasta格式.</span></span><br><span class="line"><span class="comment">#--un-conc-gz &lt;path&gt;           不能map到GENOME的reads，fasta格式, gzip压缩.</span></span><br><span class="line"><span class="comment">#--al-conc &lt;path&gt;              能map到GENOME的reads，fasta格式.</span></span><br><span class="line"><span class="comment">#--al-conc-gz &lt;path&gt;           能map到GENOME的reads，fasta格式, gzip压缩.</span></span><br><span class="line"><span class="comment">#-p/--threads NTHREADS         设置线程数. Default: 1  如果你的计算机有多个CPU或者CPU内核，那么请使用-p参数。-p参数会让bowtie进入多线程模式。每一个线程都会使用单独的CPU或者CPU内核。这种并行的运算模式也会大大加快运算速度。</span></span><br><span class="line"></span><br><span class="line">示例：</span><br><span class="line">bowtie2 -p <span class="number">10</span> -x /data/ref/bowtie2/mm10/mm10 <span class="number">-1</span> input_1.fq <span class="number">-2</span> input_2.fq | samtools sort -O bam -@ <span class="number">10</span> -o - &gt; output.bam</span><br><span class="line"><span class="comment">##双端测序数据去宿主：</span></span><br><span class="line">bowtie2 --end-to-end --no-mixed --no-discordant --no-unal --sensitive --threads <span class="number">8</span> \</span><br><span class="line">        -x /data/hostClean/Index/host.fa \</span><br><span class="line">        <span class="number">-1</span> /data/hospital_microbiome/trimmed_reads/ERR320975-trimmed-pair1.fastq.gz \</span><br><span class="line">        <span class="number">-2</span> /data/hospital_microbiome/trimmed_reads/ERR320975-trimmed-pair2.fastq.gz \</span><br><span class="line">        --un-conc-gz /data/hospital_microbiome/cleanData/ERR320975.clean.fastq.gz \</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">将sam转换成bam</span><br><span class="line"><span class="comment">#samtools view -S CJ1.sam -b &gt;CJ1.bam</span></span><br><span class="line">把bam按照默认的染色体位置排序</span><br><span class="line"><span class="comment">#samtools sort /media/wwkong/YZU_kong/osa.cold/P101SC17100126_shuidao_20180410（转录组原始数据）/IGV_data/test_bam/CJ1.bam -o /media/wwkong/YZU_kong/osa.cold/P101SC17100126_shuidao_20180410（转录组原始数据）/IGV_data/test_bam/CJ1_sorted.bam</span></span><br><span class="line">把排序文件建立索引(生成.bai文件)</span><br><span class="line"><span class="comment">#samtools index xxx_sorted.bam</span></span><br></pre></td></tr></table></figure>

<h3 id="tophat2"><a href="#tophat2" class="headerlink" title="tophat2"></a>tophat2</h3><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br></pre></td><td class="code"><pre><span class="line">用法</span><br><span class="line">tophat  [options]* &lt;genome_index_base&gt;  &lt;reads1_1[,...,readsN_1]&gt;  [reads1_2,...readsN_2]</span><br><span class="line"></span><br><span class="line">示例：</span><br><span class="line">tophat2 -p <span class="number">8</span> -G /home/cuckoo/data/liyan/train/genes.gtf -o filename_thout/ /home/cuckoo/data/liyan/train/bowtie2index/genome  filename.fq</span><br><span class="line"></span><br><span class="line">参数说明：</span><br><span class="line"><span class="comment">#-p ：指定线程数，默认为1</span></span><br><span class="line"><span class="comment">#-G ：指定已有的基因组注释信息，gtf或gff文件；</span></span><br><span class="line"><span class="comment">#-o ：指定输出目录，默认为”./tophat_out“；</span></span><br><span class="line"><span class="comment">#后面加上索引文件：与前面的bowtie2建立的索引相对应，只取前缀名。</span></span><br><span class="line"><span class="comment">#最后加上fastq文件：filename.fq；如果是双端测序则是filename_1.fq和filename_2.fq 两个文件。</span></span><br><span class="line"></span><br><span class="line">参数：</span><br><span class="line"><span class="comment">#&lt;genome_index_base&gt;  即index中的索引文件名（该文件先在bowtie中建立索引）。先在当前目录中查找索引文件，然后查找当前运行bowtie可执行文件所在目录下的indexes子目录，最后查找在自定义的环境变量BOWTIE_INDEXES（或BOWTIE2_INDEXES）中指定的目录。   建议将要建立索引的基因组序列（reference，FASTA文件）与Bowtie索引文件（index）存在于同一目录中，并且名称为&lt;genome_index_base&gt; .fa。 如果不存在，TopHat将从Bowtie索引文件中自动重建此FASTA文件。</span></span><br><span class="line"><span class="comment">#&lt;reads1_1[,...,readsN_1]&gt; 包含FASTQ或FASTA格式的reads的文件，多文件可用逗号隔开。</span></span><br><span class="line"><span class="comment">#&lt;[reads1_2,...readsN_2]&gt; 包含FASTQ或FASTA格式的reads的文件，多文件可用逗号隔开。仅当用tophat处理paired-end reads并且含有*_2的文件时出现，保证文件1与文件2的顺序相同</span></span><br><span class="line"></span><br><span class="line">Options: 操作项</span><br><span class="line"><span class="comment"># -h/--help 显示帮组文档的信息，并退出终端 </span></span><br><span class="line"><span class="comment"># -v/--version 显示tophat的版本号，并退出终端</span></span><br><span class="line"><span class="comment"># -N/--read-mismatches 丢弃错误匹配碱基数超过该数目的比对结果，默认值为2</span></span><br><span class="line"><span class="comment"># --read-gap-length 丢弃gap总长度超过该数目的比对结果，默认值为2</span></span><br><span class="line"><span class="comment"># --read-edit-dist 丢弃read的edit distance大于该值的比对结果。默认值为2 </span></span><br><span class="line"><span class="comment"># --read-realign-edit-dist 一些跨越多个外显子的reads可能会被错误地比对到geneome上。Tophat有多个比对步骤，每个比对步骤过后，比对结果中包含了edit distance的值。该参数能让Tophat对那些edit distance的值大于等于该参数的reads重新进行比对。若设置该参数值为0，则每个read在多个比对步骤中每次都要进行比对。这样会加大地增加比对精确性和运行时间。默认下该参数比上一个参数的值大，则表示对reads进行重新比对。</span></span><br><span class="line"><span class="comment"># --bowtie1 使用Bowtie1来代替Bowtie2进行比对。当使用colorspace reads时用到，因为只有Bowtie1支持，而Bowtie2不支持。默认为bowtie2</span></span><br><span class="line"><span class="comment"># -o/--output-dir &lt;string&gt; 输出的文件夹路径。默认为"./tophat_out".</span></span><br><span class="line"><span class="comment"># -r/--mate-inner-dist &lt;int&gt; 成对的reads之间的平均inner距离。例如：fragments长度300bp，两端长度50bp， 则其inner距离为200bp，该值该设为200。默认值：50bp</span></span><br><span class="line"><span class="comment"># --mate-std-dev &lt;int&gt; inner距离的标准偏差。默认值：20bp</span></span><br><span class="line"><span class="comment"># -a/--min-anchor-length &lt;int&gt; read的anchor长度：该参数能设定的最小值为3；锚定在junction两边的reads长度只有都大于此值，才能用于junction的验证。默认值：8</span></span><br><span class="line"><span class="comment"># -m/--splice-mismatches &lt;int&gt; 对于一个剪切比对，其在anchor区能出现的最大的不匹配碱基数。默认值：0</span></span><br><span class="line"><span class="comment"># -i/--min-intron-length &lt;int&gt; 最小的内含子长度。Tophat会忽略比该长度要小的donor/acceptor pairs，认为该区属于外显子。默认是70.</span></span><br><span class="line"><span class="comment"># -I/--max-intron-length &lt;int&gt; 最大的内含子长度。Tophat会忽略长度大于该值的donor/acceptor pairs，除非有long read支持。默认值是500000.</span></span><br><span class="line"><span class="comment"># --max-insertion-length &lt;int&gt; 最大的插入长度。默认值是3.</span></span><br><span class="line"><span class="comment"># --max-deletion-length &lt;int&gt; 最大缺失长度，默认值是3.</span></span><br><span class="line"><span class="comment"># --solexa-quals 对fastq文件使用Solexa的碱基质量格式</span></span><br><span class="line"><span class="comment"># --solexa1.3-quals 使用Illumina GA pipeline version 1.3的碱基质量格式，即Phred64.</span></span><br><span class="line"><span class="comment"># -Q/--quals 使用单独的碱基质量文件 </span></span><br><span class="line"><span class="comment"># --integer-quals 有空格隔开的整数值来代表碱基质量。当使用 -C 参数时，该参数为默认参数。</span></span><br><span class="line"><span class="comment"># -C/--color Colorspace reads, 注意使用一个 colorspace bowtie 索引而且需要Bowtie 0.12.6 及以上.通常用法: tophat --color --quals [other options]* &lt;colorspace_index_base&gt; &lt;reads1_1[,...,readsN_1]&gt; [reads1_2,...readsN_2] &lt;quals1_1[,...,qualsN_1]&gt; [quals1_2,...qualsN_2]</span></span><br><span class="line"><span class="comment"># -p/--num-threads &lt;int&gt; 比对时使用的线程数，默认是1.</span></span><br><span class="line"><span class="comment"># -g/--max-multihits &lt;int&gt; 对于一个reads，可能会有多个比对结果，但tophat根据比对给分，最多保留的比对结果数目。如果没有--report-secondary-alignments 参数，则只会报告出最佳的比对结果。若最佳比对结果数目超过该参数值，则只随机报告出该数目的最佳比对结果；若有 --report-secondary-alignments 参数，则按得分顺序报告出比对结果，直至达到默认的数目为止。</span></span><br><span class="line"><span class="comment"># --report-secondary-alignments 默认情况下，TopHat根据比对分数（AS）报告最佳或主要比对结果。如果要输出其他或次要比对结果（这种方法最多报告20个比对，此限制可以通过使用上面的-g / - max-multihits选项更改）请使用此选项。</span></span><br><span class="line"><span class="comment"># --no-discordant 对于paired reads，仅仅报告concordant mappings。</span></span><br><span class="line"><span class="comment"># --no-mixed 对于paired reads，只报告concordant mappings 和 discordant mappings。默认上，是所有的比对结果都报告。</span></span><br><span class="line"><span class="comment"># --no-coverage-search 取消以coverage为基础来搜寻junctions，和下一个参数互斥，该参数为默认参数。 </span></span><br><span class="line"><span class="comment"># --coverage-search 确定以覆盖度为基础来搜寻junctions。以获得最大灵敏度</span></span><br><span class="line"><span class="comment"># --microexon-search 使用该参数，pipeline会尝试寻找micro-exons。仅仅在reads长度&gt;=50bp时有效。</span></span><br><span class="line"><span class="comment"># --library-type Tophat处理的reads具有链特异性。比对结果中将会有个XS标签。一般Illumina数据的library-type为 fr-unstranded。</span></span><br></pre></td></tr></table></figure>

<h2 id="转录本组装"><a href="#转录本组装" class="headerlink" title="转录本组装"></a>转录本组装</h2><h3 id="这一步是干什么的？-1"><a href="#这一步是干什么的？-1" class="headerlink" title="这一步是干什么的？"></a>这一步是干什么的？</h3><p>回帖完之后，我们的回帖信息会被输入到一个文本文件：<strong>SAM文件</strong>（二进制位bam文件）</p>
<p>sam文件有个头文件，存储着染色体的信息，还有你之前比对的指令，但这些不是我们需要了解的重点，我们需要来看看下面存储着什么？</p>
<p><img src="https://img-blog.csdnimg.cn/20200920155413361.jpg#pic_center" alt="在这里插入图片描述"><br><strong>第一列</strong>是什么？就刚刚fastq文件的第一列，就是这条reads的名字</p>
<p><strong>第二列</strong>是什么？flag？太复杂了，不记了</p>
<p><strong>第三列</strong>？染色体</p>
<p><strong>第四列</strong>？染色体的起始位置</p>
<p><strong>第五列</strong>？回帖的可信度，即回帖质量</p>
<p>第六列？第七列？看不懂，不管了</p>
<p>总的来看，<strong>所谓的sam/bam文件就是记录回帖的序列是什么，回帖上多少，回帖的质量行不行，回帖到什么位置</strong>。</p>
<p>而后就是<strong>用cufflinks或者stringtie结合注释文件gff/gtf，将转录本构建出来</strong>。   </p>
<p>那么<strong>gff/gtf</strong>是什么？简而言之，<strong>gff就是记录了这个物种在哪个位置有功能，是gene还是调控因子</strong>。     </p>
<br>

<h3 id="GFF、GTF到底是什么？"><a href="#GFF、GTF到底是什么？" class="headerlink" title="GFF、GTF到底是什么？"></a>GFF、GTF到底是什么？</h3><h4 id="我们为什么要基因注释文件？"><a href="#我们为什么要基因注释文件？" class="headerlink" title="我们为什么要基因注释文件？"></a>我们为什么要基因注释文件？</h4><p><img src="https://img-blog.csdnimg.cn/20200920155826152.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></p>
<p>我们的<strong>gene在基因组上的结构不是连续的，而是exon-intron-exon（exon=外显子，intron=内含子）分隔开的</strong>。<strong>基因要表达，首先会先发生转录过程，转录出包含intron的pre-mRNA序列，然后再进行可变剪切，加5’帽子，3’ PolyA尾巴等一系列复杂的加工过程才会形成成熟的mRNA</strong>。</p>
<p>在进行转录组序列比对，尤其是mRNA序列比对的时候，经常需要处理跨越两个exon之间的reads，所以在进行序列比对的时候往往需要对基因组有一个注释，告诉比对软件哪个位置是gene的exon，哪个地方是gene的intron，这个就是我们所说的基因注释信息。所以，<strong>它里面核心内容就是一大堆gene在基因组上的坐标，以及这个gene本身的一些属性</strong>。</p>
<h4 id="GTF-GFF都是基因注释文件"><a href="#GTF-GFF都是基因注释文件" class="headerlink" title="GTF/GFF都是基因注释文件"></a>GTF/GFF都是基因注释文件</h4><p>  GTF = General Transfer Format</p>
<p>GFF = General Feature Format</p>
<p>GFF有若干个版本，简单来说，GTF是GFF文件的其中一个版本，我们一般认为GTF文件就是GFF 2.0版本的内容。一个标准的GTF/GFF2.0文件需要包括9列内容，一个简单的示意图如下：</p>
<p><img src="https://img-blog.csdnimg.cn/20200920161715935.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 所有的列必须用TAB分隔，总共有9列内容，第9列是补充列；</span></span><br><span class="line"><span class="comment"># 补充列的内容可以为空，但是前面8列必须有内容，如果想表达空的概念，则需要用"."；</span></span><br><span class="line"></span><br><span class="line">第<span class="number">1</span>列 seqname</span><br><span class="line"><span class="comment">#染色体的名称，需要与genome FASTA文件中的染色体名对应，别一个用"chr1"一个用"Chr1"；</span></span><br><span class="line"></span><br><span class="line">第<span class="number">2</span>列 source</span><br><span class="line"><span class="comment">#注释来自哪里，比如图2表示来自NCBI RefSeq数据库；</span></span><br><span class="line"></span><br><span class="line">第<span class="number">3</span>列 feature</span><br><span class="line"><span class="comment">#此行的注释类型，一般有exon，CDS，stop_codon, start_codon等等；</span></span><br><span class="line"></span><br><span class="line">第<span class="number">4</span>，<span class="number">5</span>列 start，end</span><br><span class="line"><span class="comment">#此行注释的起始和终止位置，标准的GTF/GFF都是以1为染色体的起点（1-based system）;</span></span><br><span class="line"><span class="comment">#注意！无论这个gene是正链还是负链，start的坐标都小于end坐标；</span></span><br><span class="line"></span><br><span class="line">第<span class="number">6</span>列 score</span><br><span class="line"><span class="comment">#一般存放打分值，比如拼装的可信度之类。下载的官方注释文件一般为0.0</span></span><br><span class="line"></span><br><span class="line">第<span class="number">7</span>列 strand</span><br><span class="line"><span class="comment">#正链基因标记为 "+", 负链基因标记为 "-";</span></span><br><span class="line"></span><br><span class="line">第<span class="number">8</span>列 frame</span><br><span class="line"><span class="comment">#只可能是0,1,2这3个值,表示与CDS中codon的相对位置；</span></span><br><span class="line"><span class="comment">#0表示，这个region的第1bp就是正好是codon 三连密码子的第1个碱基；</span></span><br><span class="line"><span class="comment">#1表示，这个region的第2bp就是正好是codon 三连密码子的第1个碱基；</span></span><br><span class="line"><span class="comment">#2表示，这个region的第3bp就是正好是codon 三连密码子的第1个碱基；</span></span><br><span class="line"></span><br><span class="line">第<span class="number">9</span>列 attribute</span><br><span class="line"><span class="comment">#一般会记录 gene_id 与transcript_id;</span></span><br><span class="line"><span class="comment">#这一列是可选列，可以增加很多内容。在程序处理过程中，相同的attribute会合并在一起处理。</span></span><br><span class="line"><span class="comment">#比如，所有gene_id=SGIP1的行都会先汇总在一起，表示1个基因。</span></span><br></pre></td></tr></table></figure>

<h3 id="Cufflinks"><a href="#Cufflinks" class="headerlink" title="Cufflinks"></a>Cufflinks</h3><p>Cufflinks程序主要根据Tophat的比对结果，依托或不依托于参考基因组的GTF注释文件，计算出(各个gene的)isoform的FPKM值，并给出trascripts.gtf注释结果(组装出转录组)</p>
<p>注意：   </p>
<blockquote>
<ol>
<li><p>fragment的长度的估测，若为pair-end测序，则cufflinks自己会有一套算法，算出结果。若为single-end测序，则cufflinks默认的是高斯分布，或者你自己提供相关的参数设置。</p>
</li>
<li><p>cufflinks计算multi-mapped reads，一般a read map到10个位置，则每个位置记为10%</p>
</li>
<li><p>一般不推荐用cufflinks拼接细菌的转录组，推荐 Glimmer。但是，若有注释文件，可以用cufflinks和cuffdiff来检测基因的表达和差异性。</p>
</li>
<li><p>cufflinks/cuffdiff不能计算出exon或splicing event的FPKM</p>
</li>
<li><p>cuffdiff处理时间序列data：采用参数-t</p>
</li>
<li><p>当你使用cufflinks时，在最后出现了99%，然后一直不动。因为cuffdiff需要更多的CPU来处理一些匹配很多reads的loci。而这些位点一般要等其他位点全部解决了后，才由cuffdiff来处理。可以用参数-M来提供相关的文件，过滤掉rRNA或者线粒体RNA。</p>
</li>
<li><p>当使用cufflinks或cuffdiff出现了“crash with a ‘bad_alloc’ error”，cuffdiff和cufflinks运行了很长时间才结束————这表明计算机拼接一个高表达的基因或定量分析一个高表达的基因，运行的内存使用玩尽了！解决方法：修改选项“-max-bundle-frags”，可以先尝试500000，若错误依旧在，可以继续下调！</p>
</li>
<li><p>cuffdiff报道的结果里面所有的基因和转录本的FPKM=0，这表明GTF中的染色体名字和BAM里的名字不匹配。</p>
</li>
<li><p>cuffdiff和cufflinks的缺点：存在一定的假基因和转录本（原因：测序深度，测序质量，测序样本的测序次数，以及注释的错误）</p>
</li>
<li><p>large fold change表达量不代表数据的明显性（这些基因的isform多或这些基因测序测到的少，整体较低的表达）。cuffdiff中明显表达倍数改变的基因，存在不确定性。</p>
</li>
<li><p>通过cufflinks产生的结果中transcript.gtf文件中cuff标识的转录本就是新的转录本。相应的，其他模块输出中CUFF标识代表着新的转录本。</p>
</li>
<li><p>若出现了如下错误：<br>You are using Cufflinks v2.2.1, which is the most recent release.<br>open: No such file or directory<br>File 30 doesn’t appear to be a valid BAM file, trying SAM…<br>Error: cannot open alignment file 30 for reading<br>这表明，你的参数有问题。例如“–min-intron-length”,你设置为了：“-min-intron-length”</p>
</li>
</ol>
</blockquote>
<p>bowtie2做的事情是:<br><img src="https://img-blog.csdnimg.cn/20200920162352791.jpg#pic_center" alt="在这里插入图片描述"><br>而cufflinks做的事情则是这个:</p>
<p><img src="https://img-blog.csdnimg.cn/20200920162420521.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br></pre></td><td class="code"><pre><span class="line">普通参数</span><br><span class="line"><span class="comment">#  -h | --help </span></span><br><span class="line"><span class="comment">#  -o | --output-dir   default: ./设置输出的文件夹名称 </span></span><br><span class="line"><span class="comment">#  -p | --num-threads  default: 1 用于比对reads的CPU线程数 </span></span><br><span class="line"><span class="comment">#  -G | --GTF 提供一个GFF文件，以此来计算isoform的表达。此时，将不会组装新的transcripts，程序会忽略和reference transcript不兼容的比对结果 </span></span><br><span class="line"><span class="comment">#  -g | --GTF-guide 提供GFF文件，以此来指导转录子组装(RABT assembly)。此时，输出结果会包含reference transcripts和novel genes and isforms。 </span></span><br><span class="line"><span class="comment">#  -M | --mask-file 提供GFF文件。Cufflinks将忽略比对到该GTF文件的transcripts中的reads。该文件中常常是rRNA的注释，也可以包含线立体和其它希望忽略的transcripts的注释。将这些不需要的RNA去除后，对计算mRNA的表达量是有利的。 </span></span><br><span class="line"><span class="comment">#  -b | --frag-bias-correct 提供一个fasta文件来指导Cufflinks运行新的bias detection and correction algorithm。这样能明显提高转录子丰度计算的精确性。 </span></span><br><span class="line"><span class="comment">#  -u | --multi-read-correct 让Cufflinks来做initial estimation步骤，从而更精确衡量比对到genome多个位点的reads。 </span></span><br><span class="line"><span class="comment">#  --library-type  default:fr-unstranded 处理的reads具有链特异性。比对结果中将会有个XS标签。一般Illumina数据的library-type为 fr-unstranded。</span></span><br><span class="line"><span class="comment">#  --library-norm-method    具体参考官网,三种方式：classic-fpkm  默认的方式。geometric  针对DESeq。quartile  计算时，fragments和总的map的count取75%</span></span><br><span class="line"></span><br><span class="line">丰度评估参数</span><br><span class="line"><span class="comment">#  -m | --frag-len-mean default: 200 插入片段的平均长度。不过现在Cufflinks能learns插入片段的平均长度，因此不推荐自主设置此值。 </span></span><br><span class="line"><span class="comment">#  -s | --frag-len-std-dev default: 80 插入片段长度的标准差。不过现在Cufflinks能learns插入片段的平均长度，因此不推荐自主设置此值。 </span></span><br><span class="line"><span class="comment">#  -N | --upper-quartile-form 使用75%分为数的值来代替总的值(比对到单一位点的fragments的数值)，作normalize。这样有利于在低丰度基因和转录子中寻找差异基因。 </span></span><br><span class="line"><span class="comment">#  --total-hits-norm default: TRUE Cufflinks在计算FPKM时,算入所有的fragments和比对上的reads。和下一个参数对立。默认激活该参数。 </span></span><br><span class="line"><span class="comment">#  --compatible-hits-norm  Cufflinks在计算FPKM时，只针对和reference transcripts兼容的fragments以及比对上的reads。该参数默认不激活，只能在有 --GTF 参数下有效，并且作 RABT或 ab initio 的时候无效。</span></span><br><span class="line"><span class="comment">#  --max-mle-iterations   进行极大似然法时选择的迭代次数，默认为：5000</span></span><br><span class="line"><span class="comment">#  --max-bundle-frags   一个skipped locus/loci在别skipped前可以拥有的最大的fragment片段。默认为1000000 </span></span><br><span class="line"><span class="comment">#  --no-effective-length-correction   Cufflinks will not employ its "effective" length normalization to transcript FPKM.Cufflinks将不会使用它的“effective” 长度标准化去计算转录的FPKM</span></span><br><span class="line"><span class="comment">#  --no-length-correction   Cufflinks将根本不会使用转录本的长度去标准化fragment的数目。当fragment的数目和the features being quantified的size是独立的，可以使用（例如for small RNA libraries, where no fragmentation takes place, or 3 prime end sequencing, where sampled RNA fragments are all essentially the same length).小心使用</span></span><br><span class="line"></span><br><span class="line">组装常用参数</span><br><span class="line"><span class="comment">#  -L | --label  default: CUFF Cufflink以GTF格式来报告转录子片段(transfrags),该参数是GTF文件的前缀 </span></span><br><span class="line"><span class="comment">#  -F/--min-isoform-fraction &lt;0.0-1.0&gt;  在计算一个基因的isoform 丰度后，过滤了丰度极低的转录本，因为这些转录本不可以信任。也可以过滤一些read匹配极低的外显子。默认为0.1或者10% of the most abundant isoform (the major isoform) of the gene.（一个基因的主要isoform的丰度的10%）</span></span><br><span class="line"><span class="comment">#  -j/--pre-mrna-fraction &lt;0.0-1.0&gt;   内含子被aligment覆盖的最低深度。若小于这个值则那些内含子的alignments被忽略掉。默认为15%。 The minimum depth of coverage in the intronic region covered      by the alignment is divided by the number of spliced reads, and if the          result is lower than this parameter value, the intronic alignments are          ignored. The default is 15%.</span></span><br><span class="line"><span class="comment">#  -I/--max-intron-length   内含子的最大长度。若大于该值的内含子，cufflinks不会报告。默认为300000.Cufflinks will not report transcripts with    introns longer than this, and will ignore SAM alignments with REF_SKIP          CIGAR operations longer than this.  The default is 300,000.</span></span><br><span class="line"><span class="comment">#  -a/--junc-alpha &lt;0.0-1.0&gt;    剪接比对过滤中假阳性的二项检验中的 alpha value。默认为 0.001</span></span><br><span class="line"><span class="comment">#  -A/--small-anchor-fraction &lt;0.0-1.0&gt;  在junction中一个reads小于自身长度的这个百分比，会被怀疑，可能会在拼接前被过滤掉。默认为0.09</span></span><br><span class="line"><span class="comment">#  --min-frags-per-transfrag   default: 10 组装出的transfrags被支持的RNA-seq的fragments数少于该值则不被报道。 </span></span><br><span class="line"><span class="comment">#  --overhang-tolerance   当决定一个reads或转录本与某个转录本兼容或匹配的时候，允许的能加入该转录本的外显子的延伸长度。默认是8bp和bowtie/tophat默认的一致。</span></span><br><span class="line"><span class="comment">#  --max-bundle-length   Maximum genomic length allowed for a given bundle.  The default is 3,500,000bp.</span></span><br><span class="line"><span class="comment">#  --min-intron-length   default: 50 最小的intron大小。 </span></span><br><span class="line"><span class="comment">#  --trim-3-avgcov-thresh   最小的3‘端的平均覆盖程度。小于该值，则删除其3’端序列。默认10  Minimum average coverage required to attempt 3' trimming.  The default is 10.</span></span><br><span class="line"><span class="comment">#  --trim-3-dropoff-frac    最低百分比的拼接的转录本的3‘端的平均覆盖程度。默认0.1  The fraction of average coverage below which to trim the 3' end of an assembled          transcript.  The default is 0.1.</span></span><br><span class="line"><span class="comment">#  --max-multiread-fraction &lt;0.0-1.0&gt;   若一个转录本Transfrags的reads能匹配到基因组的多个位置，其中该转录本的reads有超过该百分比是multireads，则不会报告这个转录本。默认为75%   The fraction a transfrag's supporting reads that may be multiply mapped to the genome. A transcript composed of more than this fraction will not be reported by the assembler.  Default: 0.75 (75% multireads or more is suppressed). </span></span><br><span class="line"><span class="comment">#  --overlap-radius   default: 50 Transfrags之间的距离少于该值，则将其连到一起。</span></span><br><span class="line"></span><br><span class="line">示例说明：</span><br><span class="line">cufflinks -p <span class="number">8</span> -o filename_clout filename_thout/accepted_hits.bam &gt;filename_cufflinks.log</span><br><span class="line"><span class="comment">#参数说明： -p ：指定线程数；-o ：指定输出文件所在目录；后面跟上Tophat2中生成的bam文件</span></span><br></pre></td></tr></table></figure>

<hr>
<h1 id="下游分析"><a href="#下游分析" class="headerlink" title="下游分析"></a>下游分析</h1><p>当使用cufflinks构建得到raw count之后，我们就想比较不同处理间的差异在哪，那么这个时候我们可以<strong>直接比较</strong>吗？ <strong>当然不可以！</strong></p>
<blockquote>
<p>问题1: 比如我有gene3，有1000条测序reads，gene4有2000条测序reads，那么我能否说gene4就一定比gene3的表达量高？    </p>
</blockquote>
<blockquote>
<p>问题2: 比如我有gene1，有1000条测序reads，我的另一个处理条件下gene2有2000条测序reads，我能否就说geneA在处理条件下表达量降低了？   </p>
</blockquote>
<p>很明显，第一个问题，<strong>如果两个基因的长度不一致，那是无法直接比较的</strong>；而第二个问题，我们就需要<strong>考虑如何矫正</strong>了，而这个矫正值就是所谓的<strong>RPKM/FPKM/TPM</strong>.</p>
<h2 id="RNA数据的定量"><a href="#RNA数据的定量" class="headerlink" title="RNA数据的定量"></a>RNA数据的定量</h2><h3 id="RNA-Seq定量过程中的比较问题？"><a href="#RNA-Seq定量过程中的比较问题？" class="headerlink" title="RNA-Seq定量过程中的比较问题？"></a>RNA-Seq定量过程中的比较问题？</h3><p><strong>RNA-Seq的基本假设是什么？</strong><br>简单来说就是 <strong>细胞/组织/个体</strong> 的两种不同状态进行比较，比较的<strong>目的就是寻找差异表达gene，然后从差异表达gene来推断造成生理状态不同的原因</strong>。</p>
<p>而我们的RNA-Seq一般情况下是针对mRNA以及带polyA的lncRNA进行建库测序分析的。那么理论上把测序的FASTQ文件mapping到参考基因组上，再结合参考基因组的GTF/GFF文件就可以找到全基因组的每一个gene上mapping到了多少个reads count。</p>
<p>拿到了reads count以后，我们就会尝试着想要比较gene之间的表达量的关系，但是这时候往往会面临两个问题，举个例子：</p>
<blockquote>
<p>问题1: 比如我有gene3，有1000条测序reads，gene4有2000条测序reads，那么我能否说gene4就一定比gene3的表达量高？(gene3&amp;gene4)<br> <br><br>问题2: 比如我有gene1，有1000条测序reads，我的另一个处理条件下gene2有2000条测序reads，我能否就说geneA在处理条件下表达量降低了？(gene1&amp;gene2)</p>
</blockquote>
<p><img src="https://img-blog.csdnimg.cn/20200920171629992.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></p>
<p>在面临这些比较问题的时候，我们就需要对mapping到gene的reads count进行矫正，至少根据问题1我们知道应该在矫正的时候考虑过gene长度的问题；根据问题2，我们大概应该能够猜想到，矫正的时候应该需要考虑整体测序量的问题。到此，RPKM和FPKM这两个指标就应运而生了。</p>
<h3 id="什么是RPKM与FPKM？"><a href="#什么是RPKM与FPKM？" class="headerlink" title="什么是RPKM与FPKM？"></a>什么是RPKM与FPKM？</h3><blockquote>
<p>RPKM = Reads Per Kilobase per Million mapped reads</p>
</blockquote>
<p>假设回贴到geneA 的 reads count为 CountA，geneA的exon总长度为Len(A) Kbp，总的测序量为D兆reads，那么：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">geneA RPKM = CountA / Len(A) / D * <span class="number">10</span>^<span class="number">9</span></span><br></pre></td></tr></table></figure>


<br>

<p>那么什么是FPKM呢？先来看一下FPKM的定义：</p>
<blockquote>
<p>FPKM = Fragments Per Kilobase per Million mapped reads</p>
</blockquote>
<p>RPKM中的R指的是Reads，FPKM中的F是指Fragments，Reads都比较好理解，就是我们的测序短的片段，那么fragment是什么呢？这是以为我们现在测序一般来说都是测双端测序（paired-end sequencing），那么在mapping回参考基因组的时候就会有两条reads，分别是read1和read2，分别来源于建库打断的5’ 端和3’端。那么这2条reads就可以在参考基因组上确定1个小的片段，这个片段就叫fragment.<br><img src="https://img-blog.csdnimg.cn/20200920172015367.jpg?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"><br>所以，如果是现在最常用的双端测序，1个gene的FPKM应该等于RPKM / 2。</p>
<h3 id="RPKM-FPKM有什么优缺点？"><a href="#RPKM-FPKM有什么优缺点？" class="headerlink" title="RPKM / FPKM有什么优缺点？"></a>RPKM / FPKM有什么优缺点？</h3><p>因为现在使用Illumina测序平台，绝大多数的测序都是使用双端测序，那么基本上我们一般对gene进行定量都是使用FPKM来进行。FPKM的优点大家都很了解了，能够矫正掉gene长度以及测序深度对gene表达定量的影响，那么FPKM的缺点大家是否熟悉呢？</p>
<p>一个比较容易被人提及的问题是对于不同批次测序的结果，所有gene的FPKM的总和不是一个固定的值。比如WT 测的所有gene的FPKM总和可能是10000，treat组测到的FPKM总和可能是15000，这样对于WT和treat组之间的差异表达gene的寻找就有可能出现问题，这个时候就需要用到我们常用的另一种矫正方法TPM。</p>
<h3 id="TPM"><a href="#TPM" class="headerlink" title="TPM"></a>TPM</h3><blockquote>
<p>TPM = Transcripts Per Million</p>
</blockquote>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="number">1.</span> 先计算每一个gene的FPKM</span><br><span class="line"><span class="number">2.</span> 计算所有gene的FPKM总和sum(FPKM)</span><br><span class="line"><span class="number">3.</span> 最终gene的TPM = gene的FPKM / sum(FPKM) * <span class="number">10</span>^<span class="number">6</span></span><br></pre></td></tr></table></figure>

<p>再简单一点说明，gene的TPM就是其FPKM百分数再乘以10^6     ,     因此一个样本的TPM的总和一定是10^6.       这样做的好处就是能够把所有样本的TPM总和统一，都变成10^6。目前很多公共数据的数据都是以TPM的方式提供，比如非常著名的GTEx数据库，提供了超过10000个人组织RNA-Seq的测序表达量矩阵，其中的表达量定量方式就选择了TPM。</p>
<h3 id="RPKM-FPKM-TPM什么情况下会定量失败？"><a href="#RPKM-FPKM-TPM什么情况下会定量失败？" class="headerlink" title="RPKM/FPKM/TPM什么情况下会定量失败？"></a>RPKM/FPKM/TPM什么情况下会定量失败？</h3><p><strong>无论是RPKM还是FPKM，亦或是TPM，本质上都是gene 表达水平的相对定量结果。</strong><br>也正是因为RNA-Seq的相对定量结果，会有2种情况导致RNA-Seq定量失败.</p>
<ul>
<li><strong>第1种情况：gene表达量发生普遍变化</strong><br>我给大家举一个非常简单的例子，比如我有两个样本分别叫做A和B，其中B的每一个gene表达量都只有对应样本A中的gene表达量的一半，那么结果就是如果使用RNA-Seq对这两个样本进行定量，那么定量的结果会是一样的，也就是两者计算的RPKM/FPKM/TPM会是一样的。<br>
通过上面的例子，我们可以作出一些推论：当两个样品中的绝大多数gene都发生相同方向的倍增或者倍减关系的时候，RNA-Seq会定量失效。这个非常重要！比如在很多癌症样本与正常组织样本的比较中，癌症样本很多时候会出现很多gene的普遍升高或者普遍降低，这个时候就需要其他特殊的办法来解决这样的问题，具体的方法后面我们会介绍。不过这种现象一般比较少见，几种常见需要考虑这个问题的情况：某些特殊表达细胞体系；很多癌症的miRNA-Seq 等。</li>
<li><strong>第2种情况，超高表达量的gene表达量发生变化</strong><br>思考一下这种情况：有两类细胞，分别是cell1和cell2，真实的情况是geneA本身是一个表达量很高的gene，同时在cell2中，geneA的表达量继续升高数倍，同时其他gene的表达量几乎不变。这个时候我们对这两种情况进行RNA-Seq建库测序，结果就会是发现其他gene普遍下调，而不仅仅是geneA的上调。如图3所示，最后建库的时候，会测到非常多的geneA的reads，其他gene的reads就会相对减少，所以会造成其他gene RPKM/FPKM/TPM下降的假象。</li>
</ul>
<h2 id="差异表达"><a href="#差异表达" class="headerlink" title="差异表达"></a>差异表达</h2><p>当我们将所有的东西放同一个标准下，就可以进行比较了，而比较的时候，即肯定存在两个组才能进行比较，也就是我们的control跟treatment</p>
<p><strong>以control为标准，比较treatment，我们就知道了差异究竟在哪些基因，即所谓的差异表达</strong></p>
<br>

<p>要求：</p>
<ul>
<li>载入表达矩阵</li>
<li>设置好分组信息</li>
<li>用DEseq2进行差异分析</li>
<li>输出差异分析结果</li>
</ul>
<h3 id="载入数据（countData和colData）"><a href="#载入数据（countData和colData）" class="headerlink" title="载入数据（countData和colData）"></a>载入数据（countData和colData）</h3><ul>
<li>正式载入数据之前，先看一下数据结构data structure，因为这是我们下面一切分析的起点</li>
<li>我们需要准备2个table：一个是countData，一个是colData</li>
</ul>
<p><img src="https://img-blog.csdnimg.cn/20200920174006455.png?x-oss-process=image/watermark,type_ZmFuZ3poZW5naGVpdGk,shadow_10,text_aHR0cHM6Ly9ibG9nLmNzZG4ubmV0L3dlaXhpbl80NTA0NDc1OA==,size_16,color_FFFFFF,t_70#pic_center" alt="在这里插入图片描述"></p>
<blockquote>
<ol>
<li>countData表示的是count矩阵，行代表gene，列代表样品，中间的数字代表对应count数。colData表示sample的元数据，因为这个表提供了sample的元数据。</li>
<li>colData的行名与countData的列名一致（除去代表gene ID的第一列）</li>
</ol>
</blockquote>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br></pre></td><td class="code"><pre><span class="line">&gt; library(tidyverse)</span><br><span class="line">&gt; library(DESeq2)</span><br><span class="line">&gt; <span class="comment">#import data</span></span><br><span class="line">&gt; setwd(<span class="string">"F:/rna_seq/data/matrix"</span>)</span><br><span class="line">&gt; mycounts&lt;-read.csv(<span class="string">"readcount.csv"</span>)</span><br><span class="line">&gt; head(mycounts)</span><br><span class="line">                   X control1 control2 treat1 treat2</span><br><span class="line"><span class="number">1</span> ENSMUSG00000000001     <span class="number">1648</span>     <span class="number">2306</span>   <span class="number">2941</span>   <span class="number">2780</span></span><br><span class="line"><span class="number">2</span> ENSMUSG00000000003        <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span></span><br><span class="line"><span class="number">3</span> ENSMUSG00000000028      <span class="number">835</span>      <span class="number">950</span>   <span class="number">1366</span>   <span class="number">1051</span></span><br><span class="line"><span class="number">4</span> ENSMUSG00000000031       <span class="number">65</span>       <span class="number">83</span>     <span class="number">52</span>     <span class="number">53</span></span><br><span class="line"><span class="number">5</span> ENSMUSG00000000037       <span class="number">70</span>       <span class="number">53</span>     <span class="number">94</span>     <span class="number">66</span></span><br><span class="line"><span class="number">6</span> ENSMUSG00000000049        <span class="number">0</span>        <span class="number">3</span>      <span class="number">4</span>      <span class="number">5</span></span><br><span class="line"><span class="comment">#这里有个x，需要去除，先把第一列当作行名来处理</span></span><br><span class="line">&gt; rownames(mycounts)&lt;-mycounts[,<span class="number">1</span>]</span><br><span class="line"><span class="comment">#把带X的列删除</span></span><br><span class="line">&gt; mycounts&lt;-mycounts[,<span class="number">-1</span>]</span><br><span class="line">&gt; head(mycounts)</span><br><span class="line">                   control1 control2 treat1 treat2</span><br><span class="line">ENSMUSG00000000001     <span class="number">1648</span>     <span class="number">2306</span>   <span class="number">2941</span>   <span class="number">2780</span></span><br><span class="line">ENSMUSG00000000003        <span class="number">0</span>        <span class="number">0</span>      <span class="number">0</span>      <span class="number">0</span></span><br><span class="line">ENSMUSG00000000028      <span class="number">835</span>      <span class="number">950</span>   <span class="number">1366</span>   <span class="number">1051</span></span><br><span class="line">ENSMUSG00000000031       <span class="number">65</span>       <span class="number">83</span>     <span class="number">52</span>     <span class="number">53</span></span><br><span class="line">ENSMUSG00000000037       <span class="number">70</span>       <span class="number">53</span>     <span class="number">94</span>     <span class="number">66</span></span><br><span class="line">ENSMUSG00000000049        <span class="number">0</span>        <span class="number">3</span>      <span class="number">4</span>      <span class="number">5</span></span><br><span class="line"><span class="comment"># 这一步很关键，要明白condition这里是因子，不是样本名称；小鼠数据有对照组和处理组，各两个重复</span></span><br><span class="line">&gt; condition &lt;- factor(c(rep(<span class="string">"control"</span>,<span class="number">2</span>),rep(<span class="string">"treat"</span>,<span class="number">2</span>)), levels = c(<span class="string">"control"</span>,<span class="string">"treat"</span>))</span><br><span class="line">&gt; condition</span><br><span class="line">[<span class="number">1</span>] control control treat   treat  </span><br><span class="line">Levels: control treat</span><br><span class="line"><span class="comment">#colData也可以自己在excel做好另存为.csv格式，再导入即可</span></span><br><span class="line">&gt; colData &lt;- data.frame(row.names=colnames(mycounts), condition)</span><br><span class="line">&gt; colData</span><br><span class="line">         condition</span><br><span class="line">control1   control</span><br><span class="line">control2   control</span><br><span class="line">treat1       treat</span><br><span class="line">treat2       treat</span><br></pre></td></tr></table></figure>

<h3 id="构建dds对象-开始DESeq流程"><a href="#构建dds对象-开始DESeq流程" class="headerlink" title="构建dds对象,开始DESeq流程"></a>构建dds对象,开始DESeq流程</h3><p><strong>注释：dds=DESeqDataSet Object</strong></p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">&gt; dds &lt;- DESeqDataSetFromMatrix(mycounts, colData, design= ~ condition)</span><br><span class="line">&gt; dds &lt;- DESeq(dds)</span><br><span class="line">&gt; <span class="comment"># 查看一下dds的内容</span></span><br><span class="line">&gt; dds</span><br></pre></td></tr></table></figure>

<p>显示为：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="class"><span class="keyword">class</span>:</span> DESeqDataSet </span><br><span class="line">dim: <span class="number">6</span> <span class="number">4</span> </span><br><span class="line">metadata(<span class="number">1</span>): version</span><br><span class="line">assays(<span class="number">3</span>): counts mu cooks</span><br><span class="line">rownames(<span class="number">6</span>): ENSMUSG00000000001 ENSMUSG00000000003 ... ENSMUSG00000000037 ENSMUSG00000000049</span><br><span class="line">rowData names(<span class="number">21</span>): baseMean baseVar ... deviance maxCooks</span><br><span class="line">colnames(<span class="number">4</span>): control1 control2 treat1 treat2</span><br><span class="line">colData names(<span class="number">2</span>): condition sizeFactor</span><br></pre></td></tr></table></figure>

<h3 id="总体结果查看"><a href="#总体结果查看" class="headerlink" title="总体结果查看"></a>总体结果查看</h3><p>接下来，我们要查看treat versus control的总体结果，并根据p-value进行重新排序。利用summary命令统计显示一共多少个genes上调和下调（FDR0.1）</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">&gt; res = results(dds, contrast=c(<span class="string">"condition"</span>, <span class="string">"control"</span>, <span class="string">"treat"</span>))</span><br><span class="line"><span class="comment">#或下面命令</span></span><br><span class="line">&gt; res= results(dds)</span><br><span class="line">&gt; res = res[order(res$pvalue),]</span><br><span class="line">&gt; head(res)</span><br><span class="line">&gt; summary(res)</span><br><span class="line"><span class="comment">#所有结果先进行输出</span></span><br><span class="line">&gt; write.csv(res,file=<span class="string">"All_results.csv"</span>)</span><br><span class="line">&gt; table(res$padj&lt;<span class="number">0.05</span>)</span><br></pre></td></tr></table></figure>

<p>上述代码的结果显示：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">&gt; res = results(dds2, contrast=c(<span class="string">"condition"</span>, <span class="string">"control"</span>, <span class="string">"treat"</span>))</span><br><span class="line">&gt; res = res[order(res$pvalue),]</span><br><span class="line">&gt; head(res)</span><br><span class="line">log2 fold change (MLE): condition control vs treat </span><br><span class="line">Wald test p-value: condition control vs treat </span><br><span class="line">DataFrame <span class="keyword">with</span> <span class="number">6</span> rows <span class="keyword">and</span> <span class="number">6</span> columns</span><br><span class="line">                    baseMean log2FoldChange     lfcSE      stat       pvalue         padj</span><br><span class="line">                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;</span><br><span class="line">ENSMUSG00000003309  <span class="number">548.1926</span>       <span class="number">3.231611</span> <span class="number">0.2658125</span> <span class="number">12.157485</span> <span class="number">5.234568e-34</span> <span class="number">8.193146e-30</span></span><br><span class="line">ENSMUSG00000046323  <span class="number">404.1894</span>       <span class="number">3.067050</span> <span class="number">0.2628220</span> <span class="number">11.669687</span> <span class="number">1.820923e-31</span> <span class="number">1.425055e-27</span></span><br><span class="line">ENSMUSG00000001123  <span class="number">341.8542</span>       <span class="number">2.797485</span> <span class="number">0.2766499</span> <span class="number">10.112004</span> <span class="number">4.887441e-24</span> <span class="number">2.549941e-20</span></span><br><span class="line">ENSMUSG00000023906  <span class="number">951.9460</span>       <span class="number">2.382307</span> <span class="number">0.2510718</span>  <span class="number">9.488551</span> <span class="number">2.342684e-21</span> <span class="number">9.116395e-18</span></span><br><span class="line">ENSMUSG00000018569  <span class="number">485.4839</span>       <span class="number">3.136031</span> <span class="number">0.3312999</span>  <span class="number">9.465836</span> <span class="number">2.912214e-21</span> <span class="number">9.116395e-18</span></span><br><span class="line">ENSMUSG00000000184  <span class="number">601.0842</span>      <span class="number">-2.827750</span> <span class="number">0.3154171</span> <span class="number">-8.965112</span> <span class="number">3.099648e-19</span> <span class="number">8.085948e-16</span></span><br><span class="line">&gt; summary(res)</span><br><span class="line"></span><br><span class="line">out of <span class="number">28335</span> <span class="keyword">with</span> nonzero total read count</span><br><span class="line">adjusted p-value &lt; <span class="number">0.1</span></span><br><span class="line">LFC &gt; <span class="number">0</span> (up)     : <span class="number">445</span>, <span class="number">1.6</span>% </span><br><span class="line">LFC &lt; <span class="number">0</span> (down)   : <span class="number">625</span>, <span class="number">2.2</span>% </span><br><span class="line">outliers [<span class="number">1</span>]     : <span class="number">0</span>, <span class="number">0</span>% </span><br><span class="line">low counts [<span class="number">2</span>]   : <span class="number">12683</span>, <span class="number">45</span>% </span><br><span class="line">(mean count &lt; <span class="number">18</span>)</span><br><span class="line">[1] see 'cooksCutoff' argument of ?results</span><br><span class="line">[2] see 'independentFiltering' argument of ?results</span><br><span class="line"></span><br><span class="line">&gt; write.csv(res,file=<span class="string">"All_results.csv"</span>)</span><br><span class="line">&gt; table(res$padj&lt;<span class="number">0.05</span>)</span><br><span class="line"></span><br><span class="line">FALSE  TRUE </span><br><span class="line"><span class="number">14909</span>   <span class="number">743</span></span><br></pre></td></tr></table></figure>

<p>可见，一共445个genes上调，625个genes下调，没有离群值。padj小于0.05的共有743个。</p>
<h3 id="提取差异表达genes（DEGs）并进行gene-symbol注释"><a href="#提取差异表达genes（DEGs）并进行gene-symbol注释" class="headerlink" title="提取差异表达genes（DEGs）并进行gene symbol注释"></a>提取差异表达genes（DEGs）并进行gene symbol注释</h3><p>差异表达基因的界定很不统一，但log2FC是用的最广泛同时也是最不精确的方式，但因为其好理解所以广泛被应用尤其芯片数据处理中，记的是havard universit做过一个统计，FC=2相对比较可靠。但无论怎么，这种界定人为因素太大，过于武断。所以GSEA，WGCNA是拿全部表达数据(可以进行初步过滤)来进行分析，并且WGCNA采取软阈值的方式来挑选genes更加合理。既然挑选差异表达基因，还是采取log2FC和padj来进行。</p>
<p><strong>获取padj（p值经过多重校验校正后的值）小于0.05，表达倍数取以2为对数后大于1或者小于-1的差异表达基因。</strong></p>
<p>代码如下：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line">&gt; diff_gene_deseq2 &lt;-subset(res, padj &lt; <span class="number">0.05</span> &amp; abs(log2FoldChange) &gt; <span class="number">1</span>)</span><br><span class="line"><span class="comment">#或</span></span><br><span class="line"><span class="comment">#&gt; diff_gene_deseq2 &lt;-subset(res,padj &lt; 0.05 &amp; (log2FoldChange &gt; 1 | log2FoldChange &lt; -1))</span></span><br><span class="line">&gt; dim(diff_gene_deseq2)</span><br><span class="line">&gt; head(diff_gene_deseq2)</span><br><span class="line">&gt; write.csv(diff_gene_deseq2,file= <span class="string">"DEG_treat_vs_control.csv"</span>)</span><br></pre></td></tr></table></figure>

<p>结果显示：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">&gt; diff_gene_deseq2 &lt;-subset(res, padj &lt; <span class="number">0.05</span> &amp; abs(log2FoldChange) &gt; <span class="number">1</span>)</span><br><span class="line">&gt; dim(diff_gene_deseq2)</span><br><span class="line">[<span class="number">1</span>] <span class="number">431</span>   <span class="number">6</span></span><br><span class="line">&gt; head(diff_gene_deseq2)</span><br><span class="line">log2 fold change (MLE): condition control vs treat </span><br><span class="line">Wald test p-value: condition control vs treat </span><br><span class="line">DataFrame <span class="keyword">with</span> <span class="number">6</span> rows <span class="keyword">and</span> <span class="number">6</span> columns</span><br><span class="line">                    baseMean log2FoldChange     lfcSE      stat       pvalue         padj</span><br><span class="line">                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;</span><br><span class="line">ENSMUSG00000003309  <span class="number">548.1926</span>       <span class="number">3.231611</span> <span class="number">0.2658125</span> <span class="number">12.157485</span> <span class="number">5.234568e-34</span> <span class="number">8.193146e-30</span></span><br><span class="line">ENSMUSG00000046323  <span class="number">404.1894</span>       <span class="number">3.067050</span> <span class="number">0.2628220</span> <span class="number">11.669687</span> <span class="number">1.820923e-31</span> <span class="number">1.425055e-27</span></span><br><span class="line">ENSMUSG00000001123  <span class="number">341.8542</span>       <span class="number">2.797485</span> <span class="number">0.2766499</span> <span class="number">10.112004</span> <span class="number">4.887441e-24</span> <span class="number">2.549941e-20</span></span><br><span class="line">ENSMUSG00000023906  <span class="number">951.9460</span>       <span class="number">2.382307</span> <span class="number">0.2510718</span>  <span class="number">9.488551</span> <span class="number">2.342684e-21</span> <span class="number">9.116395e-18</span></span><br><span class="line">ENSMUSG00000018569  <span class="number">485.4839</span>       <span class="number">3.136031</span> <span class="number">0.3312999</span>  <span class="number">9.465836</span> <span class="number">2.912214e-21</span> <span class="number">9.116395e-18</span></span><br><span class="line">ENSMUSG00000000184  <span class="number">601.0842</span>      <span class="number">-2.827750</span> <span class="number">0.3154171</span> <span class="number">-8.965112</span> <span class="number">3.099648e-19</span> <span class="number">8.085948e-16</span></span><br></pre></td></tr></table></figure>

<h3 id="用bioMart对差异表达基因进行注释"><a href="#用bioMart对差异表达基因进行注释" class="headerlink" title="用bioMart对差异表达基因进行注释"></a>用bioMart对差异表达基因进行注释</h3><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">library(<span class="string">'biomaRt'</span>)</span><br><span class="line">library(<span class="string">"curl"</span>)</span><br><span class="line">mart &lt;- useDataset(<span class="string">"mmusculus_gene_ensembl"</span>, useMart(<span class="string">"ensembl"</span>))</span><br><span class="line">my_ensembl_gene_id&lt;-row.names(diff_gene_deseq2)</span><br><span class="line"><span class="comment">#listAttributes(mart)</span></span><br><span class="line">mms_symbols&lt;- getBM(attributes=c(<span class="string">'ensembl_gene_id'</span>,<span class="string">'external_gene_name'</span>,<span class="string">"description"</span>),</span><br><span class="line">                    filters = <span class="string">'ensembl_gene_id'</span>, values = my_ensembl_gene_id, mart = mart)</span><br></pre></td></tr></table></figure>

<p>结果如下：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">&gt; library(<span class="string">'biomaRt'</span>)</span><br><span class="line">&gt; library(<span class="string">"curl"</span>)</span><br><span class="line">&gt; mart &lt;- useDataset(<span class="string">"mmusculus_gene_ensembl"</span>, useMart(<span class="string">"ensembl"</span>))</span><br><span class="line">&gt; my_ensembl_gene_id&lt;-row.names(diff_gene_deseq2)</span><br><span class="line">&gt; mms_symbols&lt;- getBM(attributes=c(<span class="string">'ensembl_gene_id'</span>,<span class="string">'external_gene_name'</span>,<span class="string">"description"</span>),</span><br><span class="line">+                     filters = <span class="string">'ensembl_gene_id'</span>, values = my_ensembl_gene_id, mart = mart)</span><br><span class="line">&gt; head(mms_symbols)</span><br><span class="line">     ensembl_gene_id external_gene_name                                                                                  description</span><br><span class="line"><span class="number">1</span> ENSMUSG00000000120               Ngfr nerve growth factor receptor (TNFR superfamily, member <span class="number">16</span>) [Source:MGI Symbol;Acc:MGI:<span class="number">97323</span>]</span><br><span class="line"><span class="number">2</span> ENSMUSG00000000184              Ccnd2                                                  cyclin D2 [Source:MGI Symbol;Acc:MGI:<span class="number">88314</span>]</span><br><span class="line"><span class="number">3</span> ENSMUSG00000000276               Dgke                           diacylglycerol kinase, epsilon [Source:MGI Symbol;Acc:MGI:<span class="number">1889276</span>]</span><br><span class="line"><span class="number">4</span> ENSMUSG00000000308              Ckmt1               creatine kinase, mitochondrial <span class="number">1</span>, ubiquitous [Source:MGI Symbol;Acc:MGI:<span class="number">99441</span>]</span><br><span class="line"><span class="number">5</span> ENSMUSG00000000320             Alox12                               arachidonate <span class="number">12</span>-lipoxygenase [Source:MGI Symbol;Acc:MGI:<span class="number">87998</span>]</span><br><span class="line"><span class="number">6</span> ENSMUSG00000000708              Kat2b                           K(lysine) acetyltransferase <span class="number">2</span>B [Source:MGI Symbol;Acc:MGI:<span class="number">1343094</span>]</span><br></pre></td></tr></table></figure>

<h3 id="合并数据-res结果-mms-symbols合并成一个文件"><a href="#合并数据-res结果-mms-symbols合并成一个文件" class="headerlink" title="合并数据:res结果+mms_symbols合并成一个文件"></a>合并数据:res结果+mms_symbols合并成一个文件</h3><p>合并的话两个数据必须有共同的列名，我们先看一下</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br></pre></td><td class="code"><pre><span class="line">&gt; head(diff_gene_deseq2)</span><br><span class="line">log2 fold change (MLE): condition control vs treat </span><br><span class="line">Wald test p-value: condition control vs treat </span><br><span class="line">DataFrame <span class="keyword">with</span> <span class="number">6</span> rows <span class="keyword">and</span> <span class="number">6</span> columns</span><br><span class="line">                    baseMean log2FoldChange     lfcSE      stat       pvalue         padj</span><br><span class="line">                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;</span><br><span class="line">ENSMUSG00000003309  <span class="number">548.1926</span>       <span class="number">3.231611</span> <span class="number">0.2658125</span> <span class="number">12.157485</span> <span class="number">5.234568e-34</span> <span class="number">8.193146e-30</span></span><br><span class="line">ENSMUSG00000046323  <span class="number">404.1894</span>       <span class="number">3.067050</span> <span class="number">0.2628220</span> <span class="number">11.669687</span> <span class="number">1.820923e-31</span> <span class="number">1.425055e-27</span></span><br><span class="line">ENSMUSG00000001123  <span class="number">341.8542</span>       <span class="number">2.797485</span> <span class="number">0.2766499</span> <span class="number">10.112004</span> <span class="number">4.887441e-24</span> <span class="number">2.549941e-20</span></span><br><span class="line">ENSMUSG00000023906  <span class="number">951.9460</span>       <span class="number">2.382307</span> <span class="number">0.2510718</span>  <span class="number">9.488551</span> <span class="number">2.342684e-21</span> <span class="number">9.116395e-18</span></span><br><span class="line">ENSMUSG00000018569  <span class="number">485.4839</span>       <span class="number">3.136031</span> <span class="number">0.3312999</span>  <span class="number">9.465836</span> <span class="number">2.912214e-21</span> <span class="number">9.116395e-18</span></span><br><span class="line">ENSMUSG00000000184  <span class="number">601.0842</span>      <span class="number">-2.827750</span> <span class="number">0.3154171</span> <span class="number">-8.965112</span> <span class="number">3.099648e-19</span> <span class="number">8.085948e-16</span></span><br><span class="line">&gt; head(mms_symbols)</span><br><span class="line">     ensembl_gene_id external_gene_name</span><br><span class="line"><span class="number">1</span> ENSMUSG00000000120               Ngfr</span><br><span class="line"><span class="number">2</span> ENSMUSG00000000184              Ccnd2</span><br><span class="line"><span class="number">3</span> ENSMUSG00000000276               Dgke</span><br><span class="line"><span class="number">4</span> ENSMUSG00000000308              Ckmt1</span><br><span class="line"><span class="number">5</span> ENSMUSG00000000320             Alox12</span><br><span class="line"><span class="number">6</span> ENSMUSG00000000708              Kat2b</span><br></pre></td></tr></table></figure>

<p>可见，两个文件没有共同的列名，所以要先给’diff_gene_deseq2’添加一个‘ensembl_gene_id’的列名，方法如下:(应该有更简便的方法)</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">&gt; ensembl_gene_id&lt;-rownames(diff_gene_deseq2)</span><br><span class="line">&gt; diff_gene_deseq2&lt;-cbind(ensembl_gene_id,diff_gene_deseq2)</span><br><span class="line">&gt; colnames(diff_gene_deseq2)[<span class="number">1</span>]&lt;-c(<span class="string">"ensembl_gene_id"</span>)</span><br><span class="line">&gt; diff_name&lt;-merge(diff_gene_deseq2,mms_symbols,by=<span class="string">"ensembl_gene_id"</span>)</span><br><span class="line">&gt;head(diff_name)</span><br><span class="line"><span class="comment">#查看Akap8的情况</span></span><br><span class="line">Akap8 &lt;- diff_name[diff_name$external_gene_name==<span class="string">"Akap8"</span>,]</span><br></pre></td></tr></table></figure>

<p>中间显示过程如下：</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br></pre></td><td class="code"><pre><span class="line">&gt; ensembl_gene_id&lt;-rownames(diff_gene_deseq2)</span><br><span class="line">&gt; diff_gene_deseq2&lt;-cbind(ensembl_gene_id,diff_gene_deseq2)</span><br><span class="line">&gt; colnames(diff_gene_deseq2)[<span class="number">1</span>]&lt;-c(<span class="string">"ensembl_gene_id"</span>)</span><br><span class="line">&gt; diff_name&lt;-merge(diff_gene_deseq2,mms_symbols,by=<span class="string">"ensembl_gene_id"</span>)</span><br><span class="line">&gt; head(diff_name)</span><br><span class="line">DataFrame <span class="keyword">with</span> <span class="number">6</span> rows <span class="keyword">and</span> <span class="number">9</span> columns</span><br><span class="line">     ensembl_gene_id   baseMean log2FoldChange     lfcSE      stat       pvalue         padj external_gene_name</span><br><span class="line">         &lt;character&gt;  &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;        &lt;character&gt;</span><br><span class="line"><span class="number">1</span> ENSMUSG00000000120  <span class="number">434.04177</span>      <span class="number">-1.293648</span> <span class="number">0.2713146</span> <span class="number">-4.768072</span> <span class="number">1.859970e-06</span> <span class="number">2.064698e-04</span>               Ngfr</span><br><span class="line"><span class="number">2</span> ENSMUSG00000000184  <span class="number">601.08417</span>      <span class="number">-2.827750</span> <span class="number">0.3154171</span> <span class="number">-8.965112</span> <span class="number">3.099648e-19</span> <span class="number">8.085948e-16</span>              Ccnd2</span><br><span class="line"><span class="number">3</span> ENSMUSG00000000276  <span class="number">668.12500</span>      <span class="number">-1.071362</span> <span class="number">0.2445381</span> <span class="number">-4.381168</span> <span class="number">1.180446e-05</span> <span class="number">9.603578e-04</span>               Dgke</span><br><span class="line"><span class="number">4</span> ENSMUSG00000000308  <span class="number">207.46719</span>       <span class="number">1.944949</span> <span class="number">0.3427531</span>  <span class="number">5.674489</span> <span class="number">1.391035e-08</span> <span class="number">3.819733e-06</span>              Ckmt1</span><br><span class="line"><span class="number">5</span> ENSMUSG00000000320   <span class="number">61.96266</span>       <span class="number">1.451927</span> <span class="number">0.4637101</span>  <span class="number">3.131109</span> <span class="number">1.741473e-03</span> <span class="number">4.105051e-02</span>             Alox12</span><br><span class="line"><span class="number">6</span> ENSMUSG00000000708 <span class="number">1070.03203</span>      <span class="number">-1.046546</span> <span class="number">0.2049062</span> <span class="number">-5.107440</span> <span class="number">3.265530e-07</span> <span class="number">5.056107e-05</span>              Kat2b</span><br><span class="line">                                                                                   description</span><br><span class="line">                                                                                   &lt;character&gt;</span><br><span class="line"><span class="number">1</span> nerve growth factor receptor (TNFR superfamily, member <span class="number">16</span>) [Source:MGI Symbol;Acc:MGI:<span class="number">97323</span>]</span><br><span class="line"><span class="number">2</span>                                                  cyclin D2 [Source:MGI Symbol;Acc:MGI:<span class="number">88314</span>]</span><br><span class="line"><span class="number">3</span>                           diacylglycerol kinase, epsilon [Source:MGI Symbol;Acc:MGI:<span class="number">1889276</span>]</span><br><span class="line"><span class="number">4</span>               creatine kinase, mitochondrial <span class="number">1</span>, ubiquitous [Source:MGI Symbol;Acc:MGI:<span class="number">99441</span>]</span><br><span class="line"><span class="number">5</span>                               arachidonate <span class="number">12</span>-lipoxygenase [Source:MGI Symbol;Acc:MGI:<span class="number">87998</span>]</span><br><span class="line"><span class="number">6</span>                           K(lysine) acetyltransferase <span class="number">2</span>B [Source:MGI Symbol;Acc:MGI:<span class="number">1343094</span>]</span><br><span class="line">&gt; Akap8 &lt;- diff_name[diff_name$external_gene_name==<span class="string">"Akap8"</span>,]</span><br><span class="line">&gt; Akap8</span><br><span class="line">DataFrame <span class="keyword">with</span> <span class="number">1</span> row <span class="keyword">and</span> <span class="number">9</span> columns</span><br><span class="line">     ensembl_gene_id  baseMean log2FoldChange     lfcSE      stat       pvalue         padj external_gene_name</span><br><span class="line">         &lt;character&gt; &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;    &lt;numeric&gt;    &lt;numeric&gt;        &lt;character&gt;</span><br><span class="line"><span class="number">1</span> ENSMUSG00000024045  <span class="number">2281.053</span>      <span class="number">-1.276329</span> <span class="number">0.2428315</span> <span class="number">-5.256028</span> <span class="number">1.471996e-07</span> <span class="number">2.775865e-05</span>              Akap8</span><br><span class="line">                                                           description</span><br><span class="line">                                                           &lt;character&gt;</span><br><span class="line"><span class="number">1</span> A kinase (PRKA) anchor protein <span class="number">8</span> [Source:MGI Symbol;Acc:MGI:<span class="number">1928488</span>]</span><br></pre></td></tr></table></figure>

<p>至此，差异表达基因提取并注释完毕！</p>
<h2 id="得到差异表达基因后怎么做"><a href="#得到差异表达基因后怎么做" class="headerlink" title="得到差异表达基因后怎么做"></a>得到差异表达基因后怎么做</h2><p>不管芯片数据还是测序数据，得到的差异表达基因DEGs都是独立的基因，如果直接对这些基因分析叫<strong>单基因分析</strong>，这种分析会有很多弊端，比如：   </p>
<ul>
<li>因为噪音的存在，结果本身就是不可靠的</li>
<li>因为对DEGs个人取舍条件的不同，也会造成结果不可靠</li>
<li>工作量增大</li>
<li>只关注单个基因而忽视基因之间的相互作用，这就很难揭示具体生物学过程，或不可靠.</li>
</ul>
<p>所以要对这些DEGs进行整合分析，这也是下游分析的关键，也就是要发现这些DEGs所揭示的生物学意义。</p>
<h3 id="聚类分析"><a href="#聚类分析" class="headerlink" title="聚类分析"></a>聚类分析</h3><p>就是把表达模式相似的基因集合到一起，然后以基因集的形式进行合并分析。通俗来说，就是把一整个相对散乱的差异基因根据表达模式相似性，划分为不同的集合，也就是后续对这些集合进行分析。通常的聚类方法有K均值算法，层次聚类，自组织映射等。</p>
<h3 id="富集分析"><a href="#富集分析" class="headerlink" title="富集分析"></a>富集分析</h3><p>富集分析需要有先验知识，那就是对已知的基因有功能注释，然后对目标基因进行功能富集。因为已经有对每个基因的功能注释，所以这种富集结果比较可靠，也有利于揭示生物学问题。</p>
<p>对基因功能注释比较常用的数据库是GO和KEGG，这基本已经成为基因注释和富集分析的标配和必需。富集分析也有两种，一种是Fisher精确检验;一种是GSEA分析,GSEA可以自己定义适合自己的基因集。</p>
<blockquote>
<p>注意：GSEA分析RNA-seq数据时，要对基因表达量进行标准化处理。</p>
</blockquote>
<h3 id="共表达网络-WGCNA）"><a href="#共表达网络-WGCNA）" class="headerlink" title="共表达网络(WGCNA）"></a>共表达网络(WGCNA）</h3><h3 id="RNA-seq数据还可以进行可变剪切，基因融合等分析"><a href="#RNA-seq数据还可以进行可变剪切，基因融合等分析" class="headerlink" title="RNA-seq数据还可以进行可变剪切，基因融合等分析"></a>RNA-seq数据还可以进行可变剪切，基因融合等分析</h3>
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      
        <div class="post-tags">
          
            <a href="/tags/RNA-seq/" rel="tag"># RNA-seq</a>
          
        </div>
      

      
      
      

      
        <div class="post-nav">
          <div class="post-nav-next post-nav-item">
            
              <a href="/2020/09/11/RNA-seq那点事儿/" rel="next" title="RNA-seq那点事儿">
                <i class="fa fa-chevron-left"></i> RNA-seq那点事儿
              </a>
            
          </div>

          <span class="post-nav-divider"></span>

          <div class="post-nav-prev post-nav-item">
            
          </div>
        </div>
      

      
      
    </footer>
  </div>
  
  
  
  </article>



    <div class="post-spread">
      
    </div>
  </div>


          </div>
          

  



        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      
        <ul class="sidebar-nav motion-element">
          <li class="sidebar-nav-toc sidebar-nav-active" data-target="post-toc-wrap">
            文章目录
          </li>
          <li class="sidebar-nav-overview" data-target="site-overview-wrap">
            站点概览
          </li>
        </ul>
      

      <section class="site-overview-wrap sidebar-panel">
        <div class="site-overview">
          <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
            
              <img class="site-author-image" itemprop="image" src="/images/avatar.png" alt="Chen kun">
            
              <p class="site-author-name" itemprop="name">Chen kun</p>
              <p class="site-description motion-element" itemprop="description"></p>
          </div>

          
            <nav class="site-state motion-element">
              
                <div class="site-state-item site-state-posts">
                
                  <a href="/archives/">
                
                    <span class="site-state-item-count">22</span>
                    <span class="site-state-item-name">日志</span>
                  </a>
                </div>
              

              
                
                
                <div class="site-state-item site-state-categories">
                  <a href="/categories/index.html">
                    
                    
                      
                    
                      
                    
                      
                    
                    <span class="site-state-item-count">3</span>
                    <span class="site-state-item-name">分类</span>
                  </a>
                </div>
              

              
                
                
                <div class="site-state-item site-state-tags">
                  <a href="/tags/index.html">
                    
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                      
                    
                    <span class="site-state-item-count">11</span>
                    <span class="site-state-item-name">标签</span>
                  </a>
                </div>
              
            </nav>
          

          

          

          
          

          
          

          
            
          
          

        </div>
      </section>

      
      <!--noindex-->
        <section class="post-toc-wrap motion-element sidebar-panel sidebar-panel-active">
          <div class="post-toc">

            
              
            

            
              <div class="post-toc-content"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#什么是高通量转录组测序？"><span class="nav-number">1.</span> <span class="nav-text">什么是高通量转录组测序？</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#准备工作"><span class="nav-number">2.</span> <span class="nav-text">准备工作</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#FASTA-amp-FASTQ"><span class="nav-number">2.1.</span> <span class="nav-text">FASTA &amp; FASTQ</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#FASTQ质量值的计算方法"><span class="nav-number">2.1.1.</span> <span class="nav-text">FASTQ质量值的计算方法</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#对测序质量这些东西进行直观化？"><span class="nav-number">2.2.</span> <span class="nav-text">对测序质量这些东西进行直观化？</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#FASTQC的基本介绍"><span class="nav-number">2.2.1.</span> <span class="nav-text">FASTQC的基本介绍</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#FASTQC的报告介绍"><span class="nav-number">2.2.2.</span> <span class="nav-text">FASTQC的报告介绍</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#质控——去接头、去污染、去重复"><span class="nav-number">2.3.</span> <span class="nav-text">质控——去接头、去污染、去重复</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#fastx-Toolkit"><span class="nav-number">2.3.1.</span> <span class="nav-text">fastx Toolkit</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#cutadapt"><span class="nav-number">2.3.2.</span> <span class="nav-text">cutadapt</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Trimmomatic"><span class="nav-number">2.3.3.</span> <span class="nav-text">Trimmomatic</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Trim-Galore"><span class="nav-number">2.3.4.</span> <span class="nav-text">Trim Galore</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#举个例子"><span class="nav-number">2.3.5.</span> <span class="nav-text">举个例子</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#上游分析"><span class="nav-number">3.</span> <span class="nav-text">上游分析</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#比对回帖"><span class="nav-number">3.1.</span> <span class="nav-text">比对回帖</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#这一步是干什么的？"><span class="nav-number">3.1.1.</span> <span class="nav-text">这一步是干什么的？</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Hisat2"><span class="nav-number">3.1.2.</span> <span class="nav-text">Hisat2</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#举个例子-1"><span class="nav-number">3.1.2.1.</span> <span class="nav-text">举个例子</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Bowtie2"><span class="nav-number">3.1.3.</span> <span class="nav-text">Bowtie2</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#tophat2"><span class="nav-number">3.1.4.</span> <span class="nav-text">tophat2</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#转录本组装"><span class="nav-number">3.2.</span> <span class="nav-text">转录本组装</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#这一步是干什么的？-1"><span class="nav-number">3.2.1.</span> <span class="nav-text">这一步是干什么的？</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#GFF、GTF到底是什么？"><span class="nav-number">3.2.2.</span> <span class="nav-text">GFF、GTF到底是什么？</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#我们为什么要基因注释文件？"><span class="nav-number">3.2.2.1.</span> <span class="nav-text">我们为什么要基因注释文件？</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#GTF-GFF都是基因注释文件"><span class="nav-number">3.2.2.2.</span> <span class="nav-text">GTF/GFF都是基因注释文件</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Cufflinks"><span class="nav-number">3.2.3.</span> <span class="nav-text">Cufflinks</span></a></li></ol></li></ol></li><li class="nav-item nav-level-1"><a class="nav-link" href="#下游分析"><span class="nav-number">4.</span> <span class="nav-text">下游分析</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#RNA数据的定量"><span class="nav-number">4.1.</span> <span class="nav-text">RNA数据的定量</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#RNA-Seq定量过程中的比较问题？"><span class="nav-number">4.1.1.</span> <span class="nav-text">RNA-Seq定量过程中的比较问题？</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#什么是RPKM与FPKM？"><span class="nav-number">4.1.2.</span> <span class="nav-text">什么是RPKM与FPKM？</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#RPKM-FPKM有什么优缺点？"><span class="nav-number">4.1.3.</span> <span class="nav-text">RPKM / FPKM有什么优缺点？</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#TPM"><span class="nav-number">4.1.4.</span> <span class="nav-text">TPM</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#RPKM-FPKM-TPM什么情况下会定量失败？"><span class="nav-number">4.1.5.</span> <span class="nav-text">RPKM/FPKM/TPM什么情况下会定量失败？</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#差异表达"><span class="nav-number">4.2.</span> <span class="nav-text">差异表达</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#载入数据（countData和colData）"><span class="nav-number">4.2.1.</span> <span class="nav-text">载入数据（countData和colData）</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#构建dds对象-开始DESeq流程"><span class="nav-number">4.2.2.</span> <span class="nav-text">构建dds对象,开始DESeq流程</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#总体结果查看"><span class="nav-number">4.2.3.</span> <span class="nav-text">总体结果查看</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#提取差异表达genes（DEGs）并进行gene-symbol注释"><span class="nav-number">4.2.4.</span> <span class="nav-text">提取差异表达genes（DEGs）并进行gene symbol注释</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#用bioMart对差异表达基因进行注释"><span class="nav-number">4.2.5.</span> <span class="nav-text">用bioMart对差异表达基因进行注释</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#合并数据-res结果-mms-symbols合并成一个文件"><span class="nav-number">4.2.6.</span> <span class="nav-text">合并数据:res结果+mms_symbols合并成一个文件</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#得到差异表达基因后怎么做"><span class="nav-number">4.3.</span> <span class="nav-text">得到差异表达基因后怎么做</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#聚类分析"><span class="nav-number">4.3.1.</span> <span class="nav-text">聚类分析</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#富集分析"><span class="nav-number">4.3.2.</span> <span class="nav-text">富集分析</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#共表达网络-WGCNA）"><span class="nav-number">4.3.3.</span> <span class="nav-text">共表达网络(WGCNA）</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#RNA-seq数据还可以进行可变剪切，基因融合等分析"><span class="nav-number">4.3.4.</span> <span class="nav-text">RNA-seq数据还可以进行可变剪切，基因融合等分析</span></a></li></ol></li></ol></li></ol></div>
            

          </div>
        </section>
      <!--/noindex-->
      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <div class="copyright">&copy; <span itemprop="copyrightYear">2020</span>
  <span class="with-love">
    <i class="fa fa-user"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Chen kun</span>

  

  
</div>




  <div class="powered-by">由 <a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a> 强力驱动</div>



  <span class="post-meta-divider">|</span>



  <div class="theme-info">主题 &mdash; <a class="theme-link" target="_blank" href="https://github.com/theme-next/hexo-theme-next">NexT.Pisces</a> v6.0.3</div>




        







        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>


















  
  









  
  
    <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>
  

  
  
    <script type="text/javascript" src="/lib/three/three.min.js"></script>
  

  
  
    <script type="text/javascript" src="/lib/three/canvas_lines.min.js"></script>
  


  


  <script type="text/javascript" src="/js/src/utils.js?v=6.0.3"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=6.0.3"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=6.0.3"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=6.0.3"></script>



  
  <script type="text/javascript" src="/js/src/scrollspy.js?v=6.0.3"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=6.0.3"></script>



  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=6.0.3"></script>



  



	





  





  










  





  

  

  

  

  
  

  

  

  

  

</body>
</html>
